---
title: "05.Somalogic"
author: "Kelsey Huus"
date: "2024-07-30"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
##Packages
library(dplyr)
library(ggplot2)
#library(data.table)
library(tidyr)
library(SomaDataIO)
library(Maaslin2)
library(ggrepel)
```

```{r soma targets}
#import the selected list of 1500 metabolites with associated panel annotations
custom_panel <- read.delim("/ebio/abt3_projects2/uHEAT2/data/somalogic/finalcuratedpanelKR.txt")
custom_panel

#select the relevant columns for export and make an excel-friendly SeqID
custom_panel$SOMAmer.SeqId_ExcelFriendly <- gsub("-", "_", custom_panel$SOMAmer.SeqId, fixed=TRUE)
custom_panel <- custom_panel %>% dplyr::select(SOMAmer.SeqId, Target.Name, Human.Target.or.Analyte, 
                                               UniProtID, GeneID, SOMAmer.SeqId_ExcelFriendly)
#write.csv(custom_panel, "/ebio/abt3_projects2/uHEAT/data/SupplementalTables/S#_SomalogicCustomPanel.csv")

```

```{r import soma}
#read in the normalized data from somalogic
soma_norm <- read_adat("/ebio/abt3_projects2/uHEAT2/data/somalogic/Somalogic_Data_ANMLNorm.adat")
soma_norm
#import regular metadata
#updated: this version also excludes SERIES 2 of all double series.
meta_seq <- read.csv("/ebio/abt3_projects2/uHEAT/data/metadata/meta_seq_clean_series1.24.08.18.csv", 
                     row.names=1)

meta_seq <- meta_seq %>% filter(!is.na(fever7d_aboveavg))
length(meta_seq$SampleID)

#per-person metadata, matching
meta <- meta_seq %>% filter(!duplicated(Participant_ID)&!is.na(fever7d_aboveavg)) %>%
  dplyr::select(-c(SampleID, Relative_Day, Time0, Time01))
length(meta$Participant_ID) #final number of participants

#define DIET 3
meta$Diet3 <- ifelse(meta$Diet %in% c("Vegan", "Vegetarian"), yes="Veg", no="Omni")
meta_seq$Diet3 <- ifelse(meta_seq$Diet %in% c("Vegan", "Vegetarian"), yes="Veg", no="Omni")

#see this vignette for info on the file.
#https://rdrr.io/github/SomaLogic/SomaDataIO/f/vignettes/loading-and-wrangling.Rmd 
soma_map <- as.data.frame(attr(soma_norm, "Col.Meta"))

soma_data_norm <- as.data.frame(soma_norm)

#extract the buffer samples that I can use later to set LOD
soma_data_blanks <- soma_data_norm %>% filter(SampleType=="Buffer")
#View(soma_data_blanks) #the overall numbers do seem to be low...

#take only my samples (norm).
soma_data_norm[grep("B", soma_data_norm$SampleId),]$SampleId #this works. #H does not work - Ronald has some H's
soma_data_norm <- soma_data_norm[grep("B", soma_data_norm$SampleId),]
row.names(soma_data_norm) <- soma_data_norm$SampleId 
#before splitting the df: based on discussion with Somalogic - remove only the flagged samples, otherwise it should be okay
soma_data_norm <- soma_data_norm %>% filter(RowCheck=="PASS")
#select only the actual metabolite columns.
names(soma_data_norm)
names(soma_data_norm)[1:33] #the metadata columns.
names(soma_data_norm)[34:length(names(soma_data_norm))] #the 1500 proteins
soma_meta <- soma_data_norm[,c(1:33)]
soma_data_norm <- soma_data_norm[,c(34:length(names(soma_data_norm)))]

#inspect the type of data that we have. (*normalized)
head(as.data.frame(soma_data_norm)) #OK 
rowSums(soma_data_norm)
colSums(soma_data_norm)
sum(is.na(soma_data_norm)) #there do not appear to be any NA values.

#align with metadata file via subject ID
meta <- meta_seq %>% filter(!duplicated(Participant_ID))
soma_meta <- soma_meta %>% rename(Participant_ID=SubjectID, Soma_SampleID = SampleId)
soma_meta <- full_join(soma_meta, meta)
head(soma_meta)
soma_meta <- soma_meta %>% filter(!is.na(Soma_SampleID)) #get rid of extra samples. One extra NA row snuck in
soma_meta$Blood <- sapply(strsplit(soma_meta$Soma_SampleID,"_"), `[`, 3)

#soma map; make a column-matching name
soma_map$SomaSeqId <- paste("seq", soma_map$SeqId, sep=".")
soma_map$SomaSeqId <- gsub("-", ".", soma_map$SomaSeqId, fixed=TRUE)

##remove hybridization controls from the possible metabolites
#before removing blanks: make sure the ACTUAL blank metabolites, hybridization control elutions, are removed.
dplyr::count(soma_map, Type)
soma_map <- soma_map %>% filter(Type=="Protein")
dim(soma_data_norm)
soma_data_norm <- soma_data_norm[,which(names(soma_data_norm)%in%soma_map$SomaSeqId)]
dim(soma_data_norm) #removed those 12 non-protein samples.
soma_data_blanks <- soma_data_blanks[,which(names(soma_data_blanks)%in%soma_map$SomaSeqId)]

#write.csv(soma_map, "/ebio/abt3_projects2/uHEAT2/data/somalogic/soma_map.csv")

```

```{r soma targeted wilcoxon}
#pick simply the main cytokines
my_cytokines <- c("IL-1b", "IL-1a", "IL-2", "IL-4", "IL-6", "IL-8", "IL-10", "IL-12", "IL-18", 
                  "IL-22", "IL-17",
                  "IFN-g", "IFN-b", "TNF-a") 
my_immunoglobulins <- c("IgG", "IgA", "IgM", "IgD", "IgE")
my_markers <- c("LBP", "HPT", "FABP2", #HPT = haptoglobin isoform 2 = zonulin (gut leakiness.?) #FABP2 should also = leak
                "PGES2", "COX-2", "CRP") #LPS binding, leakiness markers, and 2 prostaglandins #and CRP
my_tlrs <- c("TLR5", "TLR3", "TLR4", "TLR2", "TLR7", "TLR9") #I added TLR9 because maybe it has to do with IL17 #and TLR3 because viruses

#matching the dataframes: double check. #use normalized data only.
identical(names(soma_data_norm), soma_map$SomaSeqId)
#Maaslin expects the opposite orientation - samples as column names.
soma_data_norm_t <- as.data.frame(t(soma_data_norm))
identical(soma_meta$Soma_SampleID, names(soma_data_norm_t))
identical(soma_map$SomaSeqId, row.names(soma_data_norm_t))
row.names(soma_meta) <- soma_meta$Soma_SampleID

#these data frames contain B1 and B2/B3.
#baseline only:
soma_meta_baseline <- soma_meta %>% filter(Blood=="B1")
soma_data_norm_baseline <- soma_data_norm[row.names(soma_data_norm)%in%soma_meta_baseline$Soma_SampleID,]
soma_data_norm_baseline_t <- as.data.frame(t(soma_data_norm_baseline))
identical(soma_meta_baseline$Soma_SampleID, names(soma_data_norm_baseline_t))
row.names(soma_meta_baseline) <- soma_meta_baseline$Soma_SampleID
#now everything matches. So it should be quite a simple matter of running Maaslin2. 
#so that fever-hi is compared to low and not v.v.
soma_meta$fever7d_aboveavg <- factor(soma_meta$fever7d_aboveavg, levels=c("lo", "hi"))

#write out cleaned baseline data for other analyses, e.g. multi omics integration
#before writing it out, transform somalogic IDs to participant IDs
#identical(row.names(soma_data_norm_baseline), soma_meta_baseline$Soma_SampleID)
#row.names(soma_data_norm_baseline) <- soma_meta_baseline$Participant_ID
#write.csv(soma_data_norm_baseline, "/ebio/abt3_projects2/uHEAT2/data/somalogic/soma_baseline_clean.csv")

#selected cytokines for the baseline samples only
soma_map_small <- soma_map %>% filter(Target %in% c(my_cytokines, my_immunoglobulins)) #start with just cytokines and Igs
soma_data_norm_t_small <- soma_data_norm_t[which(row.names(soma_data_norm_t) %in% soma_map_small$SomaSeqId),]
soma_data_norm_t_baseline_small <- soma_data_norm_baseline_t[which(row.names(soma_data_norm_baseline_t) %in% soma_map_small$SomaSeqId),]
identical(names(soma_data_norm_t_baseline_small), soma_meta_baseline$Soma_SampleID)
#how many did I test
length(soma_map_small$Target)

#multiple wilcoxens
MW.p = apply(soma_data_norm_t_baseline_small,1,
             function(x) wilcox.test(c(x)~soma_meta_baseline$fever7d_aboveavg)$p.value)
MW.coef = apply(soma_data_norm_t_baseline_small,1,
                function(x) wilcox.test(c(x)~soma_meta_baseline$fever7d_aboveavg)$statistic)
pres <- data.frame(SomaSeqId=names(MW.p), pval=MW.p, coef=MW.coef)
pres$qval <- p.adjust(pres$pval, method='fdr')
pres[order(pres$qval),] #ooh one of them is very significant!

targeted_results <- pres %>% full_join(soma_map_small)
targeted_results <- targeted_results[order(targeted_results$pval),]
head(targeted_results)

#wilcoxen, untargeted.
identical(names(soma_data_norm_baseline_t), soma_meta_baseline$Soma_SampleID)
#how many did I test
length(soma_map$Target)

#multiple wilcoxens
MW.p2 = apply(soma_data_norm_baseline_t,1,
             function(x) wilcox.test(c(x)~soma_meta_baseline$fever7d_aboveavg)$p.value)
MW.coef2 = apply(soma_data_norm_baseline_t,1,
                function(x) wilcox.test(c(x)~soma_meta_baseline$fever7d_aboveavg)$statistic)
pres2 <- data.frame(SomaSeqId=names(MW.p2), pval=MW.p2, coef=MW.coef2)
pres2$qval <- p.adjust(pres2$pval, method='fdr')
pres2[order(pres2$qval),] #ooh one of them is very significant!

untargeted_results_w <- pres2 %>% full_join(soma_map)
untargeted_results_w <- untargeted_results_w[order(untargeted_results_w$pval),]
head(untargeted_results_w)
#very similar to Maaslin2 results.

```

```{r soma maaslin2 with and without covariates}

soma_meta_baseline$fever7d_aboveavg <- factor(soma_meta_baseline$fever7d_aboveavg, levels=c("lo", "hi"))

model_fever_norm = Maaslin2(
  input_data = soma_data_norm_baseline_t, 
  input_metadata = soma_meta_baseline, 
  output = "maaslin2_output", 
  fixed_effects = c("fever7d_aboveavg"), 
  #random_effects = c("Participant_ID"), #only Baseline
  normalization="none", #do not further normalize the data as it is already
  transform="log", #but log transform
  #min_abundance=0.0001, #no reasonable min abund exists
  min_prevalence=0.25, #must be present in 25% of samples #but there are no zeroes in this data
  max_significance=0.05, 
  plot_scatter=FALSE
)
fever_res <- model_fever_norm$results
fever_res %>% filter(qval<0.1) #hey! there is a single hit! 
fever_res %>% filter(pval<0.05) #and the p values are generally better

#export
fever_res <- fever_res %>% 
  rename(SomaSeqId=feature) %>%
  full_join(soma_map) %>%
  filter(!is.na(qval))
write.csv(fever_res, "/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/Soma_maaslin2_b1_fever.csv")

#age / sex  + fever
model_fever_corr = Maaslin2(
  input_data = soma_data_norm_baseline_t, 
  input_metadata = soma_meta_baseline, 
  output = "maaslin2_output", 
  fixed_effects = c("Sex", "Age", "fever7d_aboveavg"), 
  #random_effects = c("Participant_ID"), #only Baseline
  reference=c("Sex", "m"), #male as ref for Sex
  normalization="none", #do not further normalize the data
  transform="log", #but log transform
  #min_abundance=0.0001, #no reasonable min abund exists
  min_prevalence=0.25, #must be present in 25% of samples #but there are no zeroes in this data
  max_significance=0.05, 
  plot_scatter=FALSE
)
fever_corr1 <- model_fever_corr$results
fever_corr1 %>% filter(qval<0.05&metadata=="fever7d_aboveavg") #

#export
fever_corr1 <- fever_corr1 %>% 
  rename(SomaSeqId=feature) %>%
  full_join(soma_map) %>%
  filter(!is.na(qval))
write.csv(fever_corr1, "/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/Soma_maaslin2_b1_sex_age_fever.csv")

#age / sex / diet + fever
model_fever_corr = Maaslin2(
  input_data = soma_data_norm_baseline_t, 
  input_metadata = soma_meta_baseline, 
  output = "maaslin2_output", 
  fixed_effects = c("Sex", "Age", "Diet3", "fever7d_aboveavg"), 
  #random_effects = c("Participant_ID"), #only Baseline
  reference=c("Sex", "m"), #male as ref for Sex
  normalization="none", #do not further normalize the data
  transform="log", #but log transform
  #min_abundance=0.0001, #no reasonable min abund exists
  min_prevalence=0.25, #must be present in 25% of samples #but there are no zeroes in this data
  max_significance=0.05, 
  plot_scatter=FALSE
)
fever_corr2 <- model_fever_corr$results
fever_corr2 %>% filter(qval<0.05&metadata=="fever7d_aboveavg") #

#export
fever_corr2 <- fever_corr2 %>% 
  rename(SomaSeqId=feature) %>%
  full_join(soma_map) %>%
  filter(!is.na(qval))
write.csv(fever_corr2, "/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/Soma_maaslin2_b1_sex_age_diet3_fever.csv")

#diet / age / sex / bmi + fever
model_fever_corr = Maaslin2(
  input_data = soma_data_norm_baseline_t, 
  input_metadata = soma_meta_baseline, 
  output = "maaslin2_output", 
  fixed_effects = c("Sex", "Age", "Diet3", "BMI_kg_m2", "fever7d_aboveavg"), 
  #random_effects = c("Participant_ID"), #only Baseline
  reference=c("Sex", "m"), #male as ref for Sex
  normalization="none", #do not further normalize the data
  transform="log", #but log transform
  #min_abundance=0.0001, #no reasonable min abund exists
  min_prevalence=0.25, #must be present in 25% of samples #but there are no zeroes in this data
  max_significance=0.05, 
  plot_scatter=FALSE
)
fever_corr3 <- model_fever_corr$results
fever_corr3 %>% filter(qval<0.05&metadata=="fever7d_aboveavg") #

fever_corr3_hits <- fever_corr3 %>% filter(qval<0.1&metadata=="fever7d_aboveavg") %>% 
  rename(SomaSeqId=feature) %>%
  full_join(soma_map) %>%
  filter(!is.na(qval))
fever_corr3_hits #actually, generally the same hits
write.csv(fever_corr3_hits, "/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/Soma_maaslin2_hits_norm_baseline_sex_age_diet3_bmi_fever.csv")

#export
fever_corr3 <- fever_corr3 %>% 
  rename(SomaSeqId=feature) %>%
  full_join(soma_map) %>%
  filter(!is.na(qval))
write.csv(fever_corr3, "/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/Soma_maaslin2_b1_sex_age_diet3_bmi_fever.csv")

```

```{r soma plots2 volcano no covariates}
#instead of a tile plot, do a volcano plot for the soma hits
fever_hits_raw <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/Soma_maaslin2_b1_fever.csv", row.names=1)
fever_hits_raw

mycolours <- c("deepskyblue4", "deepskyblue3", "firebrick4", "firebrick2", "grey")
names(mycolours) <- c("DOWN q<0.05", "DOWN q<0.1", "UP q<0.05", "UP q<0.1",  "NO")
fever_hits_raw$diffexpressed <- c("NO")
fever_hits_raw$diffexpressed[fever_hits_raw$coef > 0 & fever_hits_raw$qval<0.1] <- c("UP q<0.1")
fever_hits_raw$diffexpressed[fever_hits_raw$coef > 0 & fever_hits_raw$qval<0.05] <- c("UP q<0.05")
fever_hits_raw$diffexpressed[fever_hits_raw$coef < 0 & fever_hits_raw$qval<0.1] <- c("DOWN q<0.1")
fever_hits_raw$diffexpressed[fever_hits_raw$coef < 0 & fever_hits_raw$qval<0.05] <- c("DOWN q<0.05")

# Create a new column "delabel" to de, that will contain the name of genes differentially expressed 
fever_hits_raw$delabel <- ifelse(fever_hits_raw$diffexpressed!="NO",
                          yes=fever_hits_raw$Target, no=NA)


#pdf(width=6, height=5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/2C.SomaFeverVolcano.pdf")
p = ggplot(data=fever_hits_raw, aes(x=coef, y=-log10(pval), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle("Serum Proteomics: Fever Hi vs Lo") +
  scale_colour_manual(values = mycolours) + 
  geom_point(size=0.3) + 
  theme_bw() +
  guides(fill = guide_legend(override.aes = aes(label = ""))) + 
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE)
#    geom_text(size=2)
p
#dev.off()

#list all significant hits to check annotations / labels
h1 <- fever_hits_raw %>% filter(qval<0.1) %>% dplyr::select(Target, TargetFullName, EntrezGeneSymbol, pval, qval, coef, diffexpressed)
h1$Covariates <- c("None")
h1

```

```{r soma plots2 volcano sex age}
#instead of a tile plot, do a volcano plot for the soma hits
fever_hits_raw <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/Soma_maaslin2_b1_sex_age_fever.csv", row.names=1)
fever_hits_raw <- fever_hits_raw %>% filter(metadata=="fever7d_aboveavg")

mycolours <- c("deepskyblue4", "deepskyblue3", "firebrick4", "firebrick2", "grey")
names(mycolours) <- c("DOWN q<0.05", "DOWN q<0.1", "UP q<0.05", "UP q<0.1",  "NO")
fever_hits_raw$diffexpressed <- c("NO")
fever_hits_raw$diffexpressed[fever_hits_raw$coef > 0 & fever_hits_raw$qval<0.1] <- c("UP q<0.1")
fever_hits_raw$diffexpressed[fever_hits_raw$coef > 0 & fever_hits_raw$qval<0.05] <- c("UP q<0.05")
fever_hits_raw$diffexpressed[fever_hits_raw$coef < 0 & fever_hits_raw$qval<0.1] <- c("DOWN q<0.1")
fever_hits_raw$diffexpressed[fever_hits_raw$coef < 0 & fever_hits_raw$qval<0.05] <- c("DOWN q<0.05")

# Create a new column "delabel" to de, that will contain the name of genes differentially expressed 
fever_hits_raw$delabel <- ifelse(fever_hits_raw$diffexpressed!="NO",
                          yes=fever_hits_raw$EntrezGeneSymbol, no=NA)
#annotate the ApoE proteins more completely (1-4)
fever_hits_raw$delabel <- ifelse(fever_hits_raw$delabel=="APOE", 
                                 yes=gsub("Apo ", "APO", fever_hits_raw$Target, fixed=TRUE),
                                 no=fever_hits_raw$delabel)


#pdf(width=5, height=3.5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/2A.SomaVolcano.SexAgeFever.pdf")
p = ggplot(data=fever_hits_raw, aes(x=coef, y=-log10(pval), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle("Serum Proteomics") +
  scale_colour_manual(values = mycolours) + 
  geom_point(size=0.3) + 
  theme_bw() +
  guides(fill = guide_legend(override.aes = aes(label = ""))) + 
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE)
#    geom_text(size=2)
p
#dev.off()

#list all significant hits to check annotations / labels
h2 <- fever_hits_raw %>% filter(qval<0.1) %>% dplyr::select(Target, TargetFullName, EntrezGeneSymbol, pval, qval, coef, diffexpressed)
h2$Covariates <- c("SexAge")
h2

```


```{r soma plots2 volcano sex age diet}
#instead of a tile plot, do a volcano plot for the soma hits
fever_corr_hits_raw <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/Soma_maaslin2_b1_sex_age_diet3_fever.csv", row.names=1)
fever_corr_hits_raw <- fever_corr_hits_raw %>% filter(metadata=="fever7d_aboveavg")

mycolours <- c("deepskyblue4", "deepskyblue3", "firebrick4", "firebrick2", "grey")
names(mycolours) <- c("DOWN q<0.05", "DOWN q<0.1", "UP q<0.05", "UP q<0.1",  "NO")
fever_corr_hits_raw$diffexpressed <- c("NO")
fever_corr_hits_raw$diffexpressed[fever_corr_hits_raw$coef > 0 & fever_corr_hits_raw$qval<0.1] <- c("UP q<0.1")
fever_corr_hits_raw$diffexpressed[fever_corr_hits_raw$coef > 0 & fever_corr_hits_raw$qval<0.05] <- c("UP q<0.05")
fever_corr_hits_raw$diffexpressed[fever_corr_hits_raw$coef < 0 & fever_corr_hits_raw$qval<0.1] <- c("DOWN q<0.1")
fever_corr_hits_raw$diffexpressed[fever_corr_hits_raw$coef < 0 & fever_corr_hits_raw$qval<0.05] <- c("DOWN q<0.05")

# Create a new column "delabel" to de, that will contain the name of genes differentially expressed 
fever_corr_hits_raw$delabel <- ifelse(fever_corr_hits_raw$diffexpressed!="NO",
                          yes=fever_corr_hits_raw$Target, no=NA)

#pdf(width=6, height=5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/2C.SomaFeverVolcano.pdf")
p = ggplot(data=fever_corr_hits_raw, aes(x=coef, y=-log10(pval), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle("Serum Proteomics: Fever Hi vs Lo") +
  scale_colour_manual(values = mycolours) + 
  geom_point(size=0.3) + 
  theme_bw() +
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE) 
p
#dev.off()

#export a version specifically for the legend only for all volcano plots (larger points)
#pdf(width=6, height=5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/2C.SomaFeverVolcano_LEGEND.pdf")
p = ggplot(data=fever_corr_hits_raw, aes(x=coef, y=-log10(pval), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle("Serum Proteomics: Fever Hi vs Lo") +
  scale_colour_manual(values = mycolours) + 
  geom_point(size=0.3) + 
  theme_bw() +
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE) +
    guides(colour = guide_legend(override.aes = list(size=5)))
p
#dev.off()

#also, legend only: for the plots where only the q<0.05 are shown (not q<0.1)
vdata2 <- fever_corr_hits_raw %>% filter(!(diffexpressed%in%c("UP q<0.1", "DOWN q<0.1")))

#pdf(width=6, height=5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/Volcano_LEGENDONLY_05.pdf")
p = ggplot(vdata2, aes(x=coef, y=-log10(pval), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle("Serum Proteomics: Fever Hi vs Lo") +
  scale_colour_manual(values = mycolours) + 
  geom_point(size=0.3) + 
  theme_bw() +
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE) +
    guides(colour = guide_legend(override.aes = list(size=5)))
p
#dev.off()



#list all significant hits to check annotations / labels
h3 <- fever_corr_hits_raw %>% filter(qval<0.1) %>% dplyr::select(Target, TargetFullName, EntrezGeneSymbol, pval, qval, coef, diffexpressed)
h3

#are there hits that are different when diet is accounted for
setdiff(h2$Target, h3$Target) #IL10 is among those that lose significance, but ApoE etc not


h3$Covariates <- c("SexAgeDiet")
h3

identical(names(h1), names(h2))
identical(names(h2), names(h3))

soma_hits_to_export <- Reduce(rbind, list(h1, h2, h3))
#write.csv(soma_hits_to_export, "/ebio/abt3_projects2/uHEAT/data/SupplementalTables/S#_somalogic_stats.csv")

```


```{r soma plots individual}
#Plot IL18
pdata <- data.frame(IL18=soma_data_norm_baseline$seq.5661.15,
                    Soma_SampleID=row.names(soma_data_norm_baseline)) %>%
  full_join(soma_meta_baseline) %>%
  filter(!is.na(fever7d_aboveavg))
#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT2/data/somalogic/fever7d_IL18.pdf")
p <- ggplot(pdata, aes(x=fever7d_aboveavg, y=log(IL18))) + geom_boxplot(outlier.shape=NA) + geom_jitter(width=0.1, height=0)
p <- p + theme_bw()
p <- p + annotate("text", label=c("p=0.054"), x=1.5, y=9)
p 
#dev.off()

#because of IL-18, check IL18 binding protein
soma_map[grep("interleukin-18", soma_map$TargetFullName, ignore.case = TRUE),] %>% dplyr::select(Target, TargetFullName)
soma_map[grep("IL-18 BPa", soma_map$Target),]
soma_map[grep("Interleukin-37", soma_map$TargetFullName),]
#add IL37 too because I want to export it after. #and IL10 from before
pdata <- data.frame(IL18_seq.5661.15=soma_data_norm_baseline$seq.5661.15,
                    IL18BP_seq.3073.51=soma_data_norm_baseline$seq.3073.51,
                    IL37=soma_data_norm_baseline$seq.2723.9,
                    IL10=soma_data_norm_baseline$seq.13723.6,
                    Soma_SampleID=row.names(soma_data_norm_baseline)) %>%
  full_join(soma_meta_baseline) %>%
  filter(!is.na(fever7d_aboveavg))
pdata$fever7d_aboveavg <- factor(pdata$fever7d_aboveavg, levels=c("lo", "hi"))

#it's quite similar by group but ever so slightly lower in the 'high' group.
wilcox.test(pdata$IL18BP_seq.3073.51~pdata$fever7d_aboveavg) #NS

#the ratio of IL18 to IL18BP?
pdata$IL18_vs_BP <- pdata$IL18_seq.5661.15/pdata$IL18BP_seq.3073.51
wilcox.test(pdata$IL18_vs_BP~pdata$fever7d_aboveavg) #p=0.02

#cairo_pdf(width=2.5, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/5F.IL18.pdf")
p <- ggplot(pdata, aes(x=fever7d_aboveavg, y=log(IL18_vs_BP), colour=fever7d_aboveavg)) + geom_boxplot(outlier.shape=NA) + geom_jitter(width=0.1, height=0) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  scale_colour_manual(values=c("grey50", "firebrick3")) + 
  annotate("text", label=c("*"), x=1.5, y=1.4, size=5) + 
  annotate('segment', x=1, xend=2, y=1.3, yend=1.3) + 
  ylab("Serum IL-18, free (log RFU)") + xlab("∆T") + theme(legend.position = "none")
p 
#dev.off()

#pdf(width=2.5, height=3, "/ebio/abt3_projects2/uHEAT2/data/somalogic/fever7d_IL37_clean.pdf")
p <- ggplot(pdata, aes(x=fever7d_aboveavg, y=log(IL37), colour=fever7d_aboveavg)) + geom_boxplot(outlier.shape=NA) + geom_jitter(width=0.1, height=0)
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + scale_colour_manual(values=c("grey50", "firebrick3"))
p <- p + annotate("text", label=c("**"), x=1.5, y=8.81)
p <- p + annotate('segment', x=1, xend=2, y=8.78, yend=8.78)
p <- p + ylab("Serum IL-37 (log RFU)") + xlab("Fever") + theme(legend.position = "none")
p 
#dev.off()

#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/2A.IL10.pdf")
p <- ggplot(pdata, aes(x=fever7d_aboveavg, y=log(IL10), colour=fever7d_aboveavg)) + geom_boxplot(outlier.shape=NA) + geom_jitter(width=0.1, height=0)
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + scale_colour_manual(values=c("grey50", "firebrick3"))
p <- p + annotate("text", label=c("**"), x=1.5, y=8.5)
p <- p + annotate('segment', x=1, xend=2, y=8.4, yend=8.4)
p <- p + ylab("Serum IL-10 (log RFU)") + xlab("Fever") + theme(legend.position = "none")
p 
#dev.off()

#permeability markers
#Plot LBP
pdata1 <- data.frame(LBP=soma_data_norm_baseline$seq.3074.6,
                    Soma_SampleID=row.names(soma_data_norm_baseline)) %>%
  full_join(soma_meta_baseline) %>%
  filter(!is.na(fever7d_aboveavg))
#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT2/data/somalogic/fever7d_LBP.pdf")
p <- ggplot(pdata1, aes(x=fever7d_aboveavg, y=log(LBP))) + geom_boxplot(outlier.shape=NA) + geom_jitter(width=0.1, height=0)
p <- p + theme_bw()
p <- p + annotate("text", label=c("NS, p=0.41"), x=1.5, y=11.5)
p 
#dev.off()

#Plot FABP (another leakiness marker)
soma_map[grep("FABP", soma_map$Target, ignore.case = TRUE),]
pdata1 <- data.frame(FABP=soma_data_norm_baseline$seq.15385.116,
                    Soma_SampleID=row.names(soma_data_norm_baseline)) %>%
  full_join(soma_meta_baseline) %>%
  filter(!is.na(fever7d_aboveavg))
#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT2/data/somalogic/fever7d_FABP.pdf")
p <- ggplot(pdata1, aes(x=fever7d_aboveavg, y=log(FABP))) + geom_boxplot(outlier.shape=NA) + geom_jitter(width=0.1, height=0)
p <- p + theme_bw()
p <- p + annotate("text", label=c("NS, p=0.98"), x=1.5, y=9)
p 
#dev.off()

#mini heatmap summary - see original code for other verions of this
#first: select cytokines to be included in heatmap. Pick all q<0.05 Maaslin hits and p<0.05 directed hits.
hits_targeted <- targeted_results %>% filter(pval<0.1) %>% 
  dplyr::select(SomaSeqId, TargetFullName, Target, pval, qval, coef)
hits_targeted$Hit <- c("Targeted")
hits_untargeted <- fever_corr_hits
hits_untargeted <- hits_untargeted %>% filter(qval<0.1) %>%
  dplyr::select(SomaSeqId, TargetFullName, Target, pval, qval, coef)
hits_untargeted$Hit <- c("Untargeted")
#hits_heatmap <- rbind(hits_targeted, hits_untargeted) 
hits_heatmap <- hits_untargeted #take only real untargeted hits
hits_heatmap %>% filter(duplicated(SomaSeqId)) #IL10 comes up in both; so does TLR2.
hits_heatmap <- hits_heatmap %>% filter(!duplicated(SomaSeqId))

#second: summarize the somalogic data, by fever, for each of those  targets
soma_data_heatmap <- soma_data_norm_baseline[,which(names(soma_data_norm_baseline) %in% hits_heatmap$SomaSeqId)]
identical(row.names(soma_data_heatmap), soma_meta_baseline$Soma_SampleID)
soma_data_heatmap$fever7d_aboveavg <- soma_meta_baseline$fever7d_aboveavg

names(soma_data_heatmap)
soma_data_heatmap2 <- soma_data_heatmap %>% 
  pivot_longer(cols=c(1:length(names(soma_data_heatmap))-1), names_to="SomaSeqId", values_to="RFU")
soma_data_heatmap2 <- soma_data_heatmap2 %>% 
  filter(!is.na(fever7d_aboveavg)) %>%
  group_by(fever7d_aboveavg, SomaSeqId) %>%
  summarize(RFU=mean(RFU)) %>% 
  as.data.frame()
soma_data_heatmap2[order(soma_data_heatmap2$SomaSeqId),]

#re-obtain information about the hits and use it to make a plot
soma_data_heatmap3 <- soma_data_heatmap2 %>% 
  full_join(hits_heatmap) 

#normalize the RFU row-wise 
pdata <- soma_data_heatmap3 %>% group_by(SomaSeqId) %>% mutate(RFU_norm=RFU/sum(RFU))
#and then change a bunch of things aesthetically 
pdata$fever7d_aboveavg <- factor(pdata$fever7d_aboveavg, levels=c("lo", "hi"))
pdata <- pdata %>% filter(SomaSeqId!="seq.5480.49") #choose one of the two RANTES to show, they are ~identical

pdata$Target <- factor(pdata$Target, levels=pdata$Target[order(pdata[which(pdata$fever7d_aboveavg=="hi"),]$RFU_norm, decreasing = FALSE)])

#hm
pdata$TargetFullName <- factor(pdata$TargetFullName, levels=pdata$TargetFullName[order(pdata[which(pdata$fever7d_aboveavg=="hi"),]$RFU_norm,  decreasing = FALSE)])
pdata$significance <- ifelse(pdata$fever7d_aboveavg=="hi"&pdata$coef>0&pdata$qval<0.05, 
                             yes="*", no=NA)
pdata$significance <- ifelse(pdata$fever7d_aboveavg=="lo"&pdata$coef<0&pdata$qval<0.05, 
                             yes="*", no=pdata$significance)

#optionally show only the cytokines
pdata2 <- pdata[grep("Interleukin|Interferon|necrosis", pdata$TargetFullName),]

pdf(width=4, height=4, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/2B.SomaHits.pdf")
p <- ggplot(pdata, aes(x=fever7d_aboveavg, y=Target, fill=RFU_norm)) + geom_tile(colour="white")
p <- p + scale_fill_gradient(low="aliceblue", high="deepskyblue3")
p <- p + geom_text(aes(label = significance), size = 4)
p <- p + xlab("Fever") + ylab(NULL) + labs(fill='Scaled RFU')
p
dev.off()

```

```{r soma plots3 medi}
#correlation of IL37 with the MEDI data
medi <- read.csv("/ebio/abt3_projects2/uHEAT/data/Medi/Medi_reads_per_sample_24.10.22.csv", row.names=1)

pdata <- data.frame(IL18_seq.5661.15=soma_data_norm_baseline$seq.5661.15,
                    IL18BP_seq.3073.51=soma_data_norm_baseline$seq.3073.51,
                    IL37=soma_data_norm_baseline$seq.2723.9,
                    IL10=soma_data_norm_baseline$seq.13723.6,
                    ApoE=soma_data_norm_baseline$seq.2418.55,
                    Soma_SampleID=row.names(soma_data_norm_baseline)) %>%
  full_join(soma_meta_baseline) %>%
  full_join(medi) %>%
  filter(!is.na(fever7d_aboveavg))
pdata$fever7d_aboveavg <- factor(pdata$fever7d_aboveavg, levels=c("lo", "hi"))
pdata$IL18_free <- pdata$IL18_seq.5661.15 / pdata$IL18BP_seq.3073.51

#raw plant & fungal reads vs main cytokine hits
cor.test(pdata$reads_StreptophytaBasidiomycota, pdata$IL10, method='spearman') #p=0.005
cor.test(pdata$reads_StreptophytaBasidiomycota, pdata$IL37, method='spearman') #p=0.003
cor.test(pdata$reads_StreptophytaBasidiomycota, pdata$IL18_free, method='spearman') #p=0.01
cor.test(pdata$reads_StreptophytaBasidiomycota, pdata$ApoE, method='spearman') #p=0.03

#normalized (relative abundance) plant & fungal reads vs main cytokine hits
cor.test(pdata$relabund_StreptophytaBasidiomycota, pdata$IL10, method='spearman') #p=0.008 #still significant!
cor.test(pdata$relabund_StreptophytaBasidiomycota, pdata$IL37, method='spearman') #p=0.006 #still significant!
cor.test(pdata$relabund_StreptophytaBasidiomycota, pdata$IL18_free, method='spearman') #p=0.03 #still significant!
cor.test(pdata$relabund_StreptophytaBasidiomycota, pdata$IL18_seq.5661.15, method='spearman') #NS with IL18 alone
cor.test(pdata$relabund_StreptophytaBasidiomycota, pdata$ApoE, method='spearman') #p=0.055

#total food reads vs main cytokine hits
cor.test(pdata$Total_Food_Reads, pdata$IL37, method='spearman') #p=0.002 
cor.test(pdata$Total_Food_Reads, pdata$IL18_free, method='spearman') #p=0.01 
cor.test(pdata$Total_Food_relabund, pdata$IL37, method='spearman') #p=0.004 
cor.test(pdata$Total_Food_relabund, pdata$IL18_free, method='spearman') #p=0.04


#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/2C.IL37vsPlantReadsRelAbund.pdf")
p <- ggplot(pdata, aes(x=log(relabund_StreptophytaBasidiomycota), y=IL37)) + geom_point() + geom_smooth(method='lm') + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  xlab("Plant and Funal DNA (Log Rel. Abund.)") + ylab("Serum IL-37 (RFU)") + 
  annotate('text', x=-8.5, y=6000, label="p=0.007\nrho=-0.21", hjust=0)
p
#dev.off()

#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/6D.IL18vsPlantReadsRelAbund.pdf")
p <- ggplot(pdata, aes(x=log(relabund_StreptophytaBasidiomycota), y=IL18_free)) + geom_point() + geom_smooth(method='lm') +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  xlab("Plant & Fungal DNA (Log Rel. Abund.)") + ylab("Serum IL-18, free (log RFU)") + 
  annotate('text', x=-8.5, y=3, label="p=0.040\nrho=-0.16", hjust=0)
p
#dev.off()

#instead of just reads per sample: also check the main foods, specifically hibiscus
medi_foods <- read.csv("/ebio/abt3_projects2/uHEAT/data/metadata/MEDI_foods_per_person.csv", row.names=1)
medi_foods$Participant_ID <- row.names(medi_foods)

pdata <- data.frame(IL18_seq.5661.15=soma_data_norm_baseline$seq.5661.15,
                    IL18BP_seq.3073.51=soma_data_norm_baseline$seq.3073.51,
                    IL37=soma_data_norm_baseline$seq.2723.9,
                    IL10=soma_data_norm_baseline$seq.13723.6,
                    ApoE=soma_data_norm_baseline$seq.2418.55,
                    Soma_SampleID=row.names(soma_data_norm_baseline)) %>%
  full_join(soma_meta_baseline) %>%
  full_join(medi_foods) %>%
  filter(!is.na(fever7d_aboveavg))
pdata$fever7d_aboveavg <- factor(pdata$fever7d_aboveavg, levels=c("lo", "hi"))
pdata$IL18_free <- pdata$IL18_seq.5661.15 / pdata$IL18BP_seq.3073.51

#Hibiscus vs main cytokine hits
cor.test(pdata$Roselle_.plant., pdata$IL37, method='spearman') #p=0.19,rho=-0.1
cor.test(pdata$Roselle_.plant., pdata$IL18_free, method='spearman') #p=0.17, rho=-0.1
cor.test(pdata$Roselle_.plant., pdata$ApoE, method='spearman') #NS
cor.test(pdata$Roselle_.plant., pdata$IL10, method='spearman') #NS



```

```{r medi vs soma spearmans}
#spearmans, untargeted.
#add the appropriate id's to the medi table
medi <- soma_meta_baseline %>% dplyr::select(Participant_ID, Soma_SampleID) %>%
  full_join(medi) %>%
  filter(!is.na(Soma_SampleID))
identical(names(soma_data_norm_baseline_t), medi$Soma_SampleID)
#how many did I test
length(soma_map$Target)

#multiple spearman's : all blood proteins vs the relative abundance of plant/fungal DNA in stool
SP.p = apply(soma_data_norm_baseline_t,1,
             function(x) cor.test(c(x), medi$relabund_StreptophytaBasidiomycota, method='spearman')$p.value)
SP.rho = apply(soma_data_norm_baseline_t,1,
                function(x) cor.test(c(x), medi$relabund_StreptophytaBasidiomycota, method='spearman')$estimate)
pres <- data.frame(SomaSeqId=names(SP.p), pval=SP.p, rho=SP.rho)
pres$qval <- p.adjust(pres$pval, method='fdr')
pres[order(pres$qval),] #a few are quite significant

untargeted_results_s <- pres %>% full_join(soma_map)
untargeted_results_s <- untargeted_results_s[order(untargeted_results_s$pval),]
head(untargeted_results_s) #but they are not necessarily the same fever hits
untargeted_results_s %>% filter(qval<0.05) %>% View()

#multiple spearman's : all blood proteins vs the relative abundance of food DNA in stool
SP.p = apply(soma_data_norm_baseline_t,1,
             function(x) cor.test(c(x), medi$Total_Food_relabund, method='spearman')$p.value)
SP.rho = apply(soma_data_norm_baseline_t,1,
                function(x) cor.test(c(x), medi$Total_Food_relabund, method='spearman')$estimate)
pres2 <- data.frame(SomaSeqId=names(SP.p), pval=SP.p, rho=SP.rho)
pres2$qval <- p.adjust(pres2$pval, method='fdr')
pres2[order(pres2$qval),] 

untargeted_results_s2 <- pres2 %>% full_join(soma_map)
untargeted_results_s2 <- untargeted_results_s2[order(untargeted_results_s2$pval),]
head(untargeted_results_s2) #but they are not necessarily the same fever hits
untargeted_results_s2 %>% filter(qval<0.05) %>% View() #many fewer hits

```

```{r apoE vs cholesterol}
#import cholesterol data
cholesterol <- read.csv("/ebio/abt3_projects2/uHEAT2/data/cholesterol/Cholesterol_interpolations_all.csv")
cholesterol$Participant_ID <- sapply(strsplit(cholesterol$Sample,"_"), `[`, 1)
cholesterol$Participant_ID <- gsub("H", "HEAT_", cholesterol$Participant_ID)
cholesterol$Cholesterol_Type <- sapply(strsplit(cholesterol$Sample,"_"), `[`, 3)
head(cholesterol)

#rearrange
cholesterol2 <- cholesterol %>% 
  dplyr::select(c(Participant_ID, Cholesterol_Type, Plate, Cholesterol_mg.dL)) %>%
  filter(Participant_ID!="blank") %>%
  pivot_wider(names_from=Cholesterol_Type, values_from=Cholesterol_mg.dL) %>%
  rename(HDL_Cholesterol_mg.dL=HDL, LDL_Cholesterol_mg.dL=LDL) %>%
  mutate(Total_Cholesterol_mg.dL=HDL_Cholesterol_mg.dL+LDL_Cholesterol_mg.dL)
cholesterol2$Cholesterol_ratio <- cholesterol2$LDL_Cholesterol_mg.dL / cholesterol2$HDL_Cholesterol_mg.dL

#add to soma data
apoE_ids <- soma_map[grep("Apolipoprotein E", soma_map$TargetFullName),]
apoE <- soma_data_norm_baseline %>% dplyr::select(all_of(apoE_ids$SomaSeqId)) %>%
  mutate(Soma_SampleID=row.names(soma_data_norm_baseline)) %>%
  full_join(soma_meta_baseline) %>%
  full_join(cholesterol2) %>%
  filter(!is.na(fever7d_aboveavg))
head(apoE)

#check correlations bewteen cholesterol and ApoE (total)
cor.test(apoE$seq.2418.55, apoE$Total_Cholesterol_mg.dL, method='spearman') #p=0.001, rho=0.2  #=ApoE total
cor.test(apoE$seq.2418.55, apoE$LDL_Cholesterol_mg.dL, method='spearman') #p=4.6e-06, rho=0.33
cor.test(apoE$seq.2418.55, apoE$HDL_Cholesterol_mg.dL, method='spearman') #NS
cor.test(apoE$seq.2418.55, apoE$Cholesterol_ratio, method='spearman') #p=1.0e-06, rho=0.35

#check correlations bewteen cholesterol and ApoE (isoforms)
cor.test(apoE$seq.2937.10, apoE$Cholesterol_ratio, method='spearman') #p=4e-5, rho=0.3 #ApoE3
cor.test(apoE$seq.2938.55, apoE$Cholesterol_ratio, method='spearman') #p=3.2e-06, rho=0.34 #ApoE4
cor.test(apoE$seq.5312.49, apoE$Cholesterol_ratio, method='spearman') #p=0.02, rho=0.17 #ApoE2

```