---
title: "06a.Metagenomics.Diversity"
author: "Kelsey Huus"
date: "2024-07-31"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
##Packages
library(dplyr)
library(ggplot2)
#library(data.table)
library(tidyr)
library(ape)
library(nlme)

```

```{r qc read depth}
##Read Files - QC
seqstats1 <- read.table("/ebio/abt3_projects2/uHEAT/data/output_QC/reports/final/seqkit_stats.tsv", header=TRUE) 
seqstats2 <- read.table("/ebio/abt3_projects2/uHEAT/data/output_QC_pool3/reports/final/seqkit_stats.tsv", header=TRUE) 

##first QC - read depth
#pools 1-3 combined
seqstats <- full_join(seqstats1, seqstats2)
seqstats$sample_type <- seqstats$Sample
seqstats[grep("blank", seqstats$sample_type),]$sample_type <- c("blank")
seqstats[grep("mock", seqstats$sample_type),]$sample_type <- c("mock")
seqstats$sample_type <- ifelse(seqstats$sample_type%in%(c("blank", "mock")),
                               yes=seqstats$sample_type, no="feces")
seqstats$sample_no <- sapply(strsplit(seqstats$Sample,"_"), `[`, 2)
seqstats$sample_type2 <- ifelse(seqstats$sample_type=="feces",
                                yes=paste(seqstats$sample_type, seqstats$sample_no, sep=""),
                                no=seqstats$sample_type)
summary(seqstats$num_seqs)
#seqstats_filt <- seqstats %>% filter(Read==1) #already done
#seqstats_filt$num_seqs2 <- seqstats_filt$num_seqs*2 #already done

#pdf(width=6, height=5, "/ebio/abt3_projects2/uHEAT/data/plots/seq_depth_histogram_pools1-3_postqc.pdf")
p <- ggplot(seqstats, aes(x=num_seqs, fill=sample_type, colour=sample_type)) + geom_histogram(alpha=0.1)
p <- p + xlab("Number of paired reads")
p <- p + theme_bw()
p
#dev.off()

```

```{r pcoa}
#per-sample metadata #note: mocks, blanks & samples with low read depth already excluded #so are old & infected samples
#updated: this version also excludes SERIES 2 of all double series.
meta_seq <- read.csv("/ebio/abt3_projects2/uHEAT/data/metadata/meta_seq_clean_series1.24.08.18.csv", 
                     row.names=1)
length(meta_seq$SampleID)
meta_seq$fever7d_aboveavg <- factor(meta_seq$fever7d_aboveavg, levels=c("lo", "hi"))
meta_seq$Bristol_Stool_Score2 <- ifelse(meta_seq$Bristol_Stool_Score2=="bNA", yes=NA, no=meta_seq$Bristol_Stool_Score2) #correct BSS label error

#per-person metadata, matching
meta <- meta_seq %>% filter(!duplicated(Participant_ID)&!is.na(fever7d_aboveavg)) %>%
  dplyr::select(-c(SampleID, Relative_Day, Time0, Time01))
dim(meta)

#read in beta diversity from kraken
#5 M rarefying
bray5M <- read.table("/ebio/abt3_projects2/uHEAT/data/output_kraken_all/diversity2/kraken/qiime2/5000000/beta_diversity/braycurtis.tsv",
                     header=TRUE)
head(bray5M)

#samples retained
dim(bray5M) #1046 samples retained at 5M, without the mocks/pilots etc.
#but some of these are further excluded from analysis
bray5M

#use ape
PCOA <- pcoa(bray5M)
barplot(PCOA$values$Relative_eig[1:10])
PCOA$values$Relative_eig[1:2] #so I believe the first explains 15.3% and the second 6.8%
PC1 <- round((PCOA$values$Relative_eig[1]*100), 1)
PC2 <- round((PCOA$values$Relative_eig[2]*100), 1)

#plot with some metadata
bray5M_PCA <- data.frame(PCOA$vectors)
bray5M_PCA$SampleID <- row.names(bray5M_PCA)
#no need to fix names - doubles are already correctly excluded.

bray5M_PCA <- full_join(bray5M_PCA, meta_seq) %>% filter(!is.na(fever7d_aboveavg))
bray5M_PCA <- bray5M_PCA %>% filter(SampleID %in% row.names(bray5M))
bray5M_PCA <- bray5M_PCA %>% filter(SampleID %in% meta_seq$SampleID)

#actual samples retained
length(bray5M_PCA$SampleID)

#Participant ID
#pdf(width=6, height=6, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/3A.MetagenomicPCA.pdf")
p1 <-ggplot(bray5M_PCA, aes(x=Axis.1, y=Axis.2, colour=Participant_ID, label=Participant_ID)) + geom_point(size=2)
p1 <- p1 + geom_line()
p1 <- p1 + theme_bw(base_size=16)
p1 <- p1 + ggtitle("Bray Curtis - Metagenomics") + xlab(paste("PC1 (", PC1, "%)", sep="")) + ylab(paste("PC2 (", PC2, "%)", sep=""))
p1 <- p1 + theme(legend.position="none")
p1
#dev.off()

#see file "Diversity_analses.R" for other exploratory PCOAs


#country of origin - maybe??
#check by continent 

bray5M_PCA$Continent_of_origin <- case_when(bray5M_PCA$Country_of_origin %in% c("Germany", "Switzerland", "Italy", "United Kingdom") ~ "WestEurope",
                                            bray5M_PCA$Country_of_origin %in% c("Canada", "USA", "USA/Germany") ~ "NorthAmerica",
                                            bray5M_PCA$Country_of_origin %in% c("Bosnia", "Bulgaria", "Greece") ~ "EastEurope",
                                            bray5M_PCA$Country_of_origin %in% c("Afghanistan", "Bangladesh", "Iran", "Turkey", "Syria") ~ "MiddleEast",
                                            bray5M_PCA$Country_of_origin %in% c("China", "India", "Russia", "Taiwan", "Kyrgyz Republic") ~ "Asia",
                                            bray5M_PCA$Country_of_origin %in% c("Brazil", "Chile", "Mexico", "Colombia", "Columbia") ~ "SouthAmerica",
                                            bray5M_PCA$Country_of_origin %in% c("New Zealand") ~ "Oceania")

bray5M_PCA$Country_of_origin2 <- case_when(bray5M_PCA$Country_of_origin %in% c("Germany", "Switzerland", "Italy", "Greece",
                                                                               "United Kingdom", "New Zealand",
                                                                               "Canada", "USA", "USA/Germany",
                                                                               "Taiwan", "Chile") ~ "HighIncome",
                                           bray5M_PCA$Country_of_origin %in% c("Bosnia", "Bulgaria", 
                                                                               "Turkey",  "Russia",
                                                                               "China",
                                                                               "Brazil", "Mexico", "Colombia", "Columbia") ~ "UpperMiddleIncome",
                                           bray5M_PCA$Country_of_origin %in% c("Bangladesh", "Iran", "India", "Kyrgyz Republic") ~ "LowerMiddleIncome",
                                           bray5M_PCA$Country_of_origin %in% c("Afghanistan","Syria") ~ "LowIncome")

bray5M_PCA$Continent_of_origin3 <- case_when(bray5M_PCA$Continent_of_origin %in% c("WestEurope", "EastEurope")~"Europe",
                                             bray5M_PCA$Continent_of_origin %in% c("NorthAmerica", "SouthAmerica")~"Americas",
                                             bray5M_PCA$Continent_of_origin %in% c("MiddleEast", "Asia")~"Asia",
                                             bray5M_PCA$Country_of_origin %in% c("New Zealand") ~ "Oceania")


#pdf(width=6, height=6, "/ebio/abt3_projects2/uHEAT/data/plots2/PCoA_BrayCurtis5M_all_continent.pdf")
p2 <-ggplot(bray5M_PCA, aes(x=Axis.1, y=Axis.2, colour=Country_of_origin2)) + geom_point(size=2, alpha=0.4)
p2 <- p2 + theme_bw(base_size=16)
p2 <- p2 + ggtitle("Bray Curtis - Metagenomics") + xlab(paste("PC1 (", PC1, "%)", sep="")) + ylab(paste("PC2 (", PC2, "%)", sep=""))
p2 <- p2 + scale_colour_manual(values=c("orangered", "orchid3","orange", "turquoise1", "turquoise4", "springgreen", "royalblue1"))
p2
#dev.off()

# Prevotella abundance
feces_df_genus <- read.csv("/ebio/abt3_projects2/uHEAT/data/output_kraken_all/feces_df_genus.csv",
                           row.names=1)
prevotella = data.frame(t(feces_df_genus[which(row.names(feces_df_genus)=="Prevotella"),]))
prevotella$SampleID <- row.names(prevotella)
hist(prevotella$Prevotella)
prevotella$Prevotella01 <- ifelse(prevotella$Prevotella>=0.01, "yes", "no") #ppl with at least 1% Prevotella
dplyr::count(prevotella, Prevotella01)
prevotella <- prevotella %>% filter(SampleID %in% bray5M_PCA$SampleID)
bray5M_PCA <- bray5M_PCA %>% full_join(prevotella)

hist(prevotella$Prevotella)

#pdf(width=6, height=5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S3A.MetagenomicPCA_Prevotella.pdf")
p3 <-ggplot(bray5M_PCA, aes(x=Axis.1, y=Axis.2, colour=Prevotella)) + geom_point(size=2, alpha=0.5)
p3 <- p3 + theme_bw(base_size=16)
p3 <- p3 + ggtitle("Bray Curtis - Metagenomics") + xlab(paste("PC1 (", PC1, "%)", sep="")) + ylab(paste("PC2 (", PC2, "%)", sep=""))
p3
#dev.off()

#also make one coloured by fever
bray5M_PCA$fever7d_aboveavg <- factor(bray5M_PCA$fever7d_aboveavg, levels=c("lo", "hi"))
#pdf(width=6, height=5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S3B.MetagenomicPCAFever.pdf")
p4 <-ggplot(bray5M_PCA, aes(x=Axis.1, y=Axis.2, colour=fever7d_aboveavg)) + geom_point(size=2)
p4 <- p4 + geom_line(aes(group=Participant_ID))
p4 <- p4 + theme_bw(base_size=16)
p4 <- p4 + ggtitle("Bray Curtis - Metagenomics") + xlab(paste("PC1 (", PC1, "%)", sep="")) + ylab(paste("PC2 (", PC2, "%)", sep=""))
p4 <- p4 + scale_colour_manual(values=c("grey80", "firebrick3"))
p4
#dev.off()

```

```{r distance self vs other}
#Q1: distance to self versus distance to others (overall)
bray5M_meta <- bray5M %>% mutate(SampleID=row.names(bray5M))
bray5M_long <- pivot_longer(bray5M_meta,cols=c(1:length(row.names(bray5M))),names_to="SampleID2",values_to="Bray_Distance")

#and now merge with the real meta seq
bray5M_long_meta <- meta_seq %>% dplyr::select(SampleID, Participant_ID, Relative_Day) %>%
  filter(SampleID %in% bray5M_long$SampleID) %>%
  full_join(bray5M_long) 

bray5M_long_meta$Participant_ID2 <- sapply(strsplit(bray5M_long_meta$SampleID2,"_"), `[`, 1)
bray5M_long_meta$Participant_ID2 <- gsub("H", "HEAT_", bray5M_long_meta$Participant_ID2)

#now to the actual question
bray5M_q1 <- bray5M_long_meta %>% filter(Bray_Distance>0) %>% #get rid of the auto comparisons
  group_by(Participant_ID) %>% 
  dplyr::summarize(distance_self=median(Bray_Distance[Participant_ID==Participant_ID2]),
            distance_other=median(Bray_Distance[Participant_ID!=Participant_ID2]))

#are they normal?
hist(bray5M_q1$distance_self, breaks=50) #not realllly
hist(bray5M_q1$distance_other, breaks=50) #also not really

#plot and test
pdata <- pivot_longer(bray5M_q1, cols=c(2:3), names_to="Distance", values_to="Bray_Distance")
wilcox.test(pdata$Bray_Distance~pdata$Distance) #so significant

#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/3B.MetagenomicDistance.pdf")
p <- ggplot(pdata, aes(x=Distance, y=Bray_Distance)) + geom_boxplot(outlier.shape=NA, outlier.size=NA)
p <- p + geom_jitter(width=0.1, height=0)
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + ylab("Bray Curtis Dissimilarity") #+ ggtitle("Beta Diversity")
p <- p + annotate("segment", x=1, xend=2, y=1.1, yend=1.1)
p <- p + annotate("text", x=1.5, y=1.15, label="****")
p
#dev.off()

distances_dna <- pdata #save for next plot

```

```{r distance self vs other dna & rna }

distances_rna <- read.csv("/ebio/abt3_projects2/uHEAT2/data/output_LLMGP_RNA_all/kraken/qiime2/5000000/beta_diversity/RNA_braycurtis_selfvsother_summary.csv", row.names=1)
distances_dna$SampleType <- c("DNA")
distances_rna$SampleType <- c("RNA")
identical(names(distances_dna), names(distances_rna))

pdata <- rbind(distances_dna, distances_rna)
pdata$Distance <- gsub("distance_", "", pdata$Distance) #clean up for label

#pdf(width=4, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/3C.BrayCurtsDistance.DNARNA.pdf")
p <- ggplot(pdata, aes(x=Distance, y=Bray_Distance)) + geom_boxplot(outlier.shape=NA, outlier.size=NA) +
  geom_jitter(width=0.1, height=0) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  ylab("Bray Curtis Dissimilarity") + 
  annotate("segment", x=1, xend=2, y=1.1, yend=1.1) + 
  annotate("text", x=1.5, y=1.15, label="****") +
  facet_grid(.~SampleType)
p
#dev.off()

```

```{r alpha}
shannon5M <- read.table("/ebio/abt3_projects2/uHEAT/data/output_kraken_all/kraken/qiime2/5000000/alpha_diversity/shannon.tsv", header=TRUE)

#take the median or mean per participant
shannon5M$SampleID <- row.names(shannon5M)

shannon5M_meta <- full_join(shannon5M, meta_seq)
shannon5M_meta <- shannon5M_meta %>% filter(Status.=="completed")

shannon5M_sum <- shannon5M_meta %>%
  filter(!is.na(shannon_entropy)) %>%
  group_by(Participant_ID) %>%
  summarize(mean_shannon=mean(shannon_entropy), median_shannon=median(shannon_entropy))

summary(shannon5M_sum$mean_shannon)
summary(shannon5M_sum$median_shannon) #mean and median are very similar

#bristol stool score
cor.test(as.numeric(shannon5M_meta$Bristol_Stool_Score), shannon5M_meta$shannon_entropy, method='spearman') 
#controlling for id as random effect
summary(lme(shannon_entropy ~ seq_depth + Bristol_Stool_Score2, random=~1|Participant_ID, 
            data = shannon5M_meta, 
            na.action=na.exclude, method="ML")) #still significant

#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S3B.BSSgShannon.pdf")
p <- ggplot(shannon5M_meta, aes(x=as.numeric(Bristol_Stool_Score), y=shannon_entropy)) + geom_point()
p <- p + geom_smooth(method='lm') 
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + scale_x_continuous(breaks=c(seq(1, 7, by=1)))
p <- p + annotate("text", label="p=0.03\nrho=-0.07", x=1.5, y=5.25, hjust=0)
p <- p + xlab("Bristol Stool Score") + ylab("Shannon Diversity")
p
#dev.off()


shannon5M_sum %>% dplyr::select(Participant_ID, mean_shannon) %>% head()

shannon5M_sum <- full_join(shannon5M_sum, meta) #note - fever data is again a summary of first shot
dim(shannon5M_sum)

#alpha diversity baseline only for fever
shannon5M_sum2 <- shannon5M_meta %>%
  filter(!is.na(shannon_entropy)&Time01<0) %>%
  group_by(Participant_ID) %>%
  summarize(mean_shannon=mean(shannon_entropy), median_shannon=median(shannon_entropy))
shannon5M_sum2 <- full_join(shannon5M_sum2, meta)

wilcox.test(shannon5M_sum2$mean_shannon~shannon5M_sum2$fever7d_aboveavg) #NS
pdata1 <- shannon5M_sum2 %>% filter(!is.na(fever7d_aboveavg))
pdata1$fever7d_aboveavg <- factor(pdata1$fever7d_aboveavg, levels=c("lo", "hi"))

#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S3B.MetagenomicShannon.pdf")
p <- ggplot(pdata1, aes(x=fever7d_aboveavg, y=mean_shannon)) + geom_boxplot(outlier.size=NA, outlier.shape=NA)
p <- p + geom_jitter(width=0.1, height=0)
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + annotate("segment", x=1, xend=2, y=9, yend=9)
p <- p + annotate("text", label="NS", x=1.5, y=9.2)
p <- p + xlab("Fever") + ylab("Baseline Shannon Diversity")
p
#dev.off()

```


```{r distance by time}

bray5M_meta <- bray5M %>% mutate(SampleID=row.names(bray5M))
bray5M_long <- pivot_longer(bray5M_meta,cols=c(1:length(names(bray5M))),
                            names_to="SampleID2",values_to="Bray_Distance") #the comparison sample

#adjust the mixed samples
#double checking that it is already correct
meta_seq %>% filter(Participant_ID=="HEAT_051") %>% dplyr::select(Participant_ID, SampleID, sample_no)

#and now merge with the real meta seq
bray5M_long_meta <- meta_seq %>% dplyr::select(SampleID, Participant_ID, Relative_Day) %>%
  filter(SampleID %in% bray5M_long$SampleID) %>% #minus samples that were below rarefaction
  full_join(bray5M_long) %>%
  filter(SampleID %in% meta_seq$SampleID) #only keep clean samples

bray5M_long_meta$Participant_ID2 <- sapply(strsplit(bray5M_long_meta$SampleID2,"_"), `[`, 1)
bray5M_long_meta$Participant_ID2 <- gsub("H", "HEAT_", bray5M_long_meta$Participant_ID2)

#Q2: distance to self at each individual time point
#first: check which relative days are NA, because these will result in the whole person being kicked out
bray5M_long_meta %>% filter(is.na(Relative_Day)) %>% 
  filter(!duplicated(SampleID)) %>% dplyr::select(SampleID, Relative_Day)
#all possible missing relative days now added - it is clean - the rest should be excluded

bray5M_long_meta <- bray5M_long_meta %>%
  filter(!is.na(Relative_Day))

#define bray-curtis distance per day #ID==ID2, i.e., same person is being compared
#for example at first day, t-7, the data looks like this:
head(bray5M_long_meta[which(bray5M_long_meta$Relative_Day=="t-7"&bray5M_long_meta$Participant_ID==bray5M_long_meta$Participant_ID2),])

bray5M_q2 <- bray5M_long_meta %>% filter(Bray_Distance>0) %>% #get rid of the exact auto comparisons, i.e. same sample
  group_by(Participant_ID) %>% #per person
  dplyr::summarize(distance0=median(Bray_Distance[Relative_Day=="t-7"&Participant_ID==Participant_ID2]),
            distance1=median(Bray_Distance[Relative_Day=="t-6"&Participant_ID==Participant_ID2]),
            distance2=median(Bray_Distance[Relative_Day=="t-5"&Participant_ID==Participant_ID2]),
            distance3=median(Bray_Distance[Relative_Day=="t-4"&Participant_ID==Participant_ID2]),
            distance4=median(Bray_Distance[Relative_Day=="t-3"&Participant_ID==Participant_ID2]),
            distance5=median(Bray_Distance[Relative_Day=="t-2"&Participant_ID==Participant_ID2]),
            distance6=median(Bray_Distance[Relative_Day=="t-1"&Participant_ID==Participant_ID2]),
            distance7=median(Bray_Distance[Relative_Day=="t0"&Participant_ID==Participant_ID2]),
            distance8=median(Bray_Distance[Relative_Day=="t+1"&Participant_ID==Participant_ID2]),
            distance9=median(Bray_Distance[Relative_Day=="t+2"&Participant_ID==Participant_ID2]),
            distance10=median(Bray_Distance[Relative_Day=="t+3"&Participant_ID==Participant_ID2]),
            distance11=median(Bray_Distance[Relative_Day=="t+4"&Participant_ID==Participant_ID2]),
            distance12=median(Bray_Distance[Relative_Day=="t+5"&Participant_ID==Participant_ID2]),
            distance13=median(Bray_Distance[Relative_Day=="t+6"&Participant_ID==Participant_ID2]),
            distance14=median(Bray_Distance[Relative_Day=="t+7"&Participant_ID==Participant_ID2])
            
  )
head(as.data.frame(bray5M_q2)) #much sparser of course #it is still necessary to specify distance to self!!

#plot and test
pdata <- pivot_longer(bray5M_q2, cols=c(2:length(names(bray5M_q2))), 
                      names_to="Day", values_to="Bray_Distance")
pdata$Day <- gsub("distance", "d", pdata$Day)
pdata$Day <- factor(pdata$Day, levels=c("d0", "d1", "d2", "d3", "d4", "d5", "d6", "d7",
                                        "d8", "d9", "d10", "d11", "d12", "d13", "d14"))
#add fever data per participant
pdata <- pdata %>% full_join(meta) %>% filter(!is.na(Day))

#smoothed by Time01
pdata$Relative_Day_numeric <- as.numeric(gsub("d", "", pdata$Day)) - 7
brayq2_meta <- pdata %>% dplyr::select(Participant_ID, Relative_Day_numeric, Bray_Distance) %>% 
  full_join(meta_seq) #recollect per-sample information
brayq2_meta <- brayq2_meta %>% filter(!is.na(Relative_Day_numeric) & !is.na(fever7d_aboveavg))
#brayq2_meta <- brayq2_meta %>% filter(Relative_Day_numeric<=8)

#pdf("/ebio/abt3_projects2/uHEAT/data/CleanPlots/3D.BrayTemporalFever.pdf", width=4, height=3)
p <- ggplot(brayq2_meta, aes(x=Time01, y=Bray_Distance, colour=fever7d_aboveavg)) + 
  #geom_point() + 
  geom_line(aes(group=Participant_ID), alpha=0.3) +
  geom_smooth(method='loess', fill='white', alpha=0.7) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  xlab("Time (days)") + ylab("Bray Curtis Dissimilarity") + ggtitle("Distance to Self") + labs(colour="Fever") + 
  scale_x_continuous(breaks = (seq(-7, 7, by = 7))) + 
  scale_colour_manual(values=c("grey50", "firebrick3")) + 
  annotate('segment', x=0, xend=0, y=0.1, yend=0.7, linetype='dashed')
p
#dev.off()

#test statistically via mixed linear model #although there is no visual pattern
m1 <- lme(Bray_Distance ~ seq_depth + Bristol_Stool_Score2 + Time01 + fever7d_aboveavg + Time01*fever7d_aboveavg, random=~1|Participant_ID, data = brayq2_meta, 
            na.action=na.exclude, method="ML")
summary(m1) #NS


```

```{r alpha by time}
shannon5M <- read.table("/ebio/abt3_projects2/uHEAT/data/output_kraken_all/kraken/qiime2/5000000/alpha_diversity/shannon.tsv", header=TRUE)

#join with metadata
shannon5M$SampleID <- row.names(shannon5M)
shannon5M_meta <- full_join(shannon5M, meta_seq)
shannon5M_meta <- shannon5M_meta %>% filter(Status.=="completed")

#plot longitudinal alpha diversity by fever
pdata <- shannon5M_meta %>% filter(!is.na(fever7d_aboveavg)&Time01<=8)
#pdf("/ebio/abt3_projects2/uHEAT/data/CleanPlots/3E.ShannonTemporalFever.pdf", width=4, height=3)
p <- ggplot(pdata, aes(x=Time01, y=shannon_entropy, colour=fever7d_aboveavg)) + 
  #geom_point() + 
  geom_line(aes(group=Participant_ID), alpha=0.3) +
  geom_smooth(method='loess', fill='white', alpha=0.7) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  xlab("Time (days)") + ylab("Shannon entropy") + ggtitle("Alpha Diversity") + labs(colour="Fever") + 
  scale_x_continuous(breaks = (seq(-7, 7, by = 7))) + 
  scale_colour_manual(values=c("grey50", "firebrick3")) + 
  annotate('segment', x=0, xend=0, y=5, yend=9, linetype='dashed')
p
dev.off()

#longitudinal 1
pdata1 <- pdata %>% filter(!is.na(sample_no))
pdata1$sample_no3 <- case_when(pdata1$sample_no==8 ~ 1,
                               pdata1$sample_no==9 ~ 2,
                               pdata1$sample_no==10 ~ 3,
                               pdata1$sample_no==11 ~ 4,
                               pdata1$sample_no==12 ~ 5,
                               pdata1$sample_no==13 ~ 6)
pdata1$sample_no3 <- ifelse(is.na(pdata1$sample_no3), 
                            yes=pdata1$sample_no, no=pdata1$sample_no3)


#this type of plot looks cleaner, hm.
p <- ggplot(pdata1, aes(x=as.factor(sample_no3), y=shannon_entropy, colour=fever7d_aboveavg)) + geom_boxplot(outlier.size=NA, outlier.shape=NA)
p <- p + geom_point(position=position_jitterdodge(jitter.width=0.1, jitter.height=0))
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + scale_colour_manual(values=c("grey50", "firebrick3"))
#p <- p + scale_x_continuous(breaks=c(seq(-8, 8, by=1)))
#p <- p + annotate("segment", x=3.5, xend=3.5, y=-9, yend=-2, linetype='dashed')
#p <- p + annotate("text", x=3.5, y=-1.5, label=c("Vaccine"))
p <- p + xlab("Timepoint") + ylab("Shannon Entropy") + ggtitle("Alpha Diversity") + labs(colour="Fever")
p

#test statistically via mixed linear model #the same way as for beta diversity 
m1 <- lme(shannon_entropy ~ seq_depth +  Bristol_Stool_Score2 + Time01 + fever7d_aboveavg + Time01*fever7d_aboveavg, random=~1|Participant_ID, data = shannon5M_meta, 
            na.action=na.exclude, method="ML")
summary(m1) #NS


```

