---
title: "06b.Metagenomics.Maaslin"
author: "Kelsey Huus"
date: "2024-07-31"
output: html_document
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
##Packages
library(dplyr)
library(tidyr)
library(Maaslin2)
library(ggplot2)
library(nlme)

```


```{r import}
#per-sample metadata #note: mocks, blanks & samples with low read depth already excluded #so are old & infected samples
#updated: this version also excludes SERIES 2 of all double series.
meta_seq <- read.csv("/ebio/abt3_projects2/uHEAT/data/metadata/meta_seq_clean_series1.24.08.18.csv", 
                     row.names=1)
meta_seq$Bristol_Stool_Score2 <- ifelse(meta_seq$Bristol_Stool_Score2=="bNA", yes=NA, no=meta_seq$Bristol_Stool_Score2) #correct BSS label error


#number of clean samples
length(meta_seq$SampleID)

#per-person metadata, matching
meta <- meta_seq %>% filter(!duplicated(Participant_ID)&!is.na(fever7d_aboveavg)) %>%
  dplyr::select(-c(SampleID, Relative_Day, Time0, Time01))
dim(meta)

#define DIET 3
meta$Diet3 <- ifelse(meta$Diet %in% c("Vegan", "Vegetarian"), yes="Veg", no="Omni")
meta_seq$Diet3 <- ifelse(meta_seq$Diet %in% c("Vegan", "Vegetarian"), yes="Veg", no="Omni")

#import the cleaned data tables generated by "DNA.Maaslin.Executable.R"
feces_df_genus <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_input/DNA_feces_df_genus.csv", row.names=1)
feces_df_species <-  read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_input/DNA_feces_df_species.csv", row.names=1)

#check the order, filtering, etc.
feces_df_genus <- feces_df_genus[,order(names(feces_df_genus))] #order
feces_df_species <- feces_df_species[,order(names(feces_df_species))] #order
meta_seq <- meta_seq[order(meta_seq$SampleID),] #order
identical(names(feces_df_genus), meta_seq$SampleID)
identical(names(feces_df_species), meta_seq$SampleID)

#later, for plotting
plot_gen <- data.frame(t(feces_df_genus), meta_seq)
plot_spec <- data.frame(t(feces_df_species), meta_seq)

#baseline only
meta_seq_baseline <- meta_seq %>% filter(Stool_timing3=="before")
feces_df_genus_baseline <- feces_df_genus[,which(names(feces_df_genus)%in%meta_seq_baseline$SampleID)]
identical(names(feces_df_genus_baseline), meta_seq_baseline$SampleID)
feces_df_species_baseline <- feces_df_species[,which(names(feces_df_species)%in%meta_seq_baseline$SampleID)]
identical(names(feces_df_species_baseline), meta_seq_baseline$SampleID)

```

```{r maaslin2 results}

##Fever - overall (genus) #7d fever with technical covariates only

#check results
res_long_fevergen <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/DNAg_seqBSSTimeRT_fever7d.csv", row.names=1)
res_long_fevergen %>% filter(metadata=="fever7d_aboveavg"&qval<0.05) %>% as.data.frame() #Acetatifactor, OK

##Fever - overall (species) #7d fever with technical covariates only
#check results
res_long_feverspec <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/DNAs_seqBSSTimeRT_fever7d.csv", row.names=1)
res_long_feverspec %>% filter(metadata=="fever7d_aboveavg"&qval<0.05) %>% as.data.frame() #Also mostly Acetatifactor (up) + Strep/CAG245 down


#export both for supplemental table
res_export1 <- res_long_fevergen %>% filter(qval<0.1&metadata=="fever7d_aboveavg") %>% dplyr::select(feature, value, coef, pval, qval, N, N.not.zero)
res_export1$Covariates <- c("seq_depth+BSS+Time_at_RT")
res_export1$Taxonomic_level <- c("Genus")

res_export2 <- res_long_feverspec %>% filter(qval<0.1&metadata=="fever7d_aboveavg") %>% dplyr::select(feature, value, coef, pval, qval, N, N.not.zero)
res_export2$Covariates <- c("seq_depth+BSS+Time_at_RT")
res_export2$Taxonomic_level <- c("Species")

res_export <- rbind(res_export1, res_export2)

#baseline only genus
res_long_fevergenbefore <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/DNAg_before_seqBSSTimeRT_fever7d.csv", row.names=1)
res_long_fevergenbefore %>% filter(metadata=="fever7d_aboveavg"&qval<0.05) %>% as.data.frame() #nothing

#baseline only species
res_long_feverspecbefore <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/DNAs_before_seqBSSTimeRT_fever7d.csv", row.names=1)
res_long_feverspecbefore %>% filter(metadata=="fever7d_aboveavg"&qval<0.05) %>% as.data.frame() #CAG245 down

#IgG2 results for comparison: genus
res_long_igg2gen <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/Rscripts_server/maaslin2_output/DNAg_seqBSSTimeRT_IgG_B2.csv")
res_long_igg2gen %>% filter(metadata=="IgG_B2_aboveavg"&qval<0.05) %>% as.data.frame() #Metalachnospira
res_long_igg2gen %>% dplyr::filter(feature=="Acetatifactor"&metadata=="IgG_B2_aboveavg") #raw p=0.058



```


```{r maaslin2 results corr}

##Fever - overall (genus) #7d fever with additional covariates only
res_long_fevergen_corr <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/DNAg_seqBSSTimeRT_SexAgeDiet3_fever7d.csv")
res_long_fevergen_corr %>% dplyr::filter(qval<0.05&metadata=="fever7d_aboveavg") #nothing
res_long_fevergen_corr %>% dplyr::filter(qval<0.05&metadata=="Diet3") #Anaerotignum, Bariatricus, Avimicrobium 
res_long_fevergen_corr %>% dplyr::filter(feature=="Acetatifactor"&metadata=="Diet3") #NS, p=0.26

res_long_feverspec_corr <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/DNAs_seqBSSTimeRT_SexAgeDiet3_fever7d.csv")
res_long_feverspec_corr %>% dplyr::filter(qval<0.05&metadata=="fever7d_aboveavg") #CAG245 down

#export both (corrected results) for supplemental table
res_export1 <- res_long_fevergen_corr %>% filter(qval<0.1&metadata=="fever7d_aboveavg") %>% dplyr::select(feature, value, coef, pval, qval, N, N.not.zero)
res_export1$Covariates <- c("seq_depth+BSS+Time_at_RT+Age+Sex+Diet")
res_export1$Taxonomic_level <- c("Genus")

res_export2 <- res_long_feverspec_corr %>% filter(qval<0.1&metadata=="fever7d_aboveavg") %>% dplyr::select(feature, value, coef, pval, qval, N, N.not.zero)
res_export2$Covariates <- c("seq_depth+BSS+Time_at_RT+Age+Sex+Diet")
res_export2$Taxonomic_level <- c("Species")

res_export_corr <- rbind(res_export1, res_export2)
res_export <- rbind(res_export, res_export_corr)

#write.csv(res_export, "/ebio/abt3_projects2/uHEAT/data/SupplementalTables/Table#_Fig4D.csv")

#baseline only
res_long_fevergenbefore_corr <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/DNAg_before_seqBSSTimeRT_SexAgeDiet3_fever7d.csv")
res_long_fevergenbefore_corr %>% dplyr::filter(qval<0.05&metadata=="fever7d_aboveavg") #nothing

res_long_feverspecbefore_corr <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_output/DNAs_before_seqBSSTimeRT_SexAgeDiet3_fever7d.csv")
res_long_feverspecbefore_corr %>% dplyr::filter(qval<0.05&metadata=="fever7d_aboveavg") #nothing



```


```{r plots Waltera}
#genus Acetatifactor over time
plot_gen$fever7d_aboveavg <- factor(plot_gen$fever7d_aboveavg, levels=c("lo", "hi"))
plot_gen$Diet2 <- factor(plot_gen$Diet2, levels=c("Veg", "Omni"))
pdata <- plot_gen %>% filter(!is.na(fever7d_aboveavg))

pdata1 <- pdata %>% filter(!is.na(sample_no))
pdata1$sample_no3 <- case_when(pdata1$sample_no==8 ~ 1,
                               pdata1$sample_no==9 ~ 2,
                               pdata1$sample_no==10 ~ 3,
                               pdata1$sample_no==11 ~ 4,
                               pdata1$sample_no==12 ~ 5,
                               pdata1$sample_no==13 ~ 6)
pdata1$sample_no3 <- ifelse(is.na(pdata1$sample_no3), 
                            yes=pdata1$sample_no, no=pdata1$sample_no3)


#cairo_pdf(width=4, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/4D.MetagenomicAcetatifactor.pdf")
p <- ggplot(pdata1, aes(x=as.factor(sample_no3), y=log(Acetatifactor), colour=fever7d_aboveavg)) + geom_boxplot(outlier.size=NA, outlier.shape=NA) + 
  geom_point(position=position_jitterdodge(jitter.width=0.1, jitter.height=0)) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  scale_colour_manual(values=c("grey50", "firebrick3")) + 
  annotate("segment", x=3.5, xend=3.5, y=-9, yend=-2, linetype=2) + 
  annotate("text", x=3.5, y=-1.5, label=c("Vaccine")) + 
  xlab("Timepoint") + ylab("Log Rel. Abund.") + ggtitle("Waltera") + labs(colour="âˆ†T")
p
#dev.off()

#what is the overall (nonlog) relative abundance of Acetatifactor?
summary(plot_gen$Acetatifactor)
summary(plot_gen$Acetatifactor*100)
#present in all samples, from 0.008% to 11.6%, mean ~1% abundance - not bad

#any difference by diet? #yes
pdata1$Diet3 <- factor(pdata1$Diet3, levels=c("Veg", "Omni"))
#pdf(width=4, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S3D.MetagenomicAcetatifactor_Diet3.pdf")
p <- ggplot(pdata1, aes(x=as.factor(sample_no3), y=log(Acetatifactor), colour=Diet3)) + 
  geom_boxplot(outlier.size=NA, outlier.shape=NA) + 
  geom_point(position=position_jitterdodge(jitter.width=0.1, jitter.height=0)) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  scale_colour_manual(values=c("darkcyan", "lightsalmon3")) + 
  annotate("segment", x=3.5, xend=3.5, y=-9, yend=-2, linetype=2) + 
  annotate("text", x=3.5, y=-1.5, label=c("Vaccine")) + 
  xlab("Timepoint") + ylab("Log Rel. Abund.") + ggtitle("Waltera") + labs(colour="Diet")
p
#dev.off()

#longitudinal significance via mixed linear model
m1 <- lme(Acetatifactor ~ seq_depth + Bristol_Stool_Score2 + Time_at_RT + Diet3, random=~1|Participant_ID, 
            data = pdata1, 
            na.action=na.exclude, method="ML") #NS
summary(m1) #diet p=0.1466


#does Waltera associate with BSS? #no not especially
p <- ggplot(pdata1, aes(x=log(Acetatifactor), y=as.numeric(Bristol_Stool_Score))) + 
  geom_point() + 
  geom_smooth(method='lm') +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p

#Summarize Acetatifactor per person
pdata2 <- pdata %>% dplyr::group_by(Participant_ID, fever7d_aboveavg, Diet2, Diet3) %>%
  dplyr::summarize(Acetatifactor=mean(Acetatifactor)) 

#pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT/data/plots3/Fever7d_vs_Acetatifactor_sum1.pdf")
p <- ggplot(pdata2, aes(x=fever7d_aboveavg, y=log(Acetatifactor), colour=fever7d_aboveavg)) + geom_boxplot(outlier.size=NA, outlier.shape=NA)
p <- p + geom_jitter(width=0.1, height=0)
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + scale_colour_manual(values=c("grey50", "firebrick3"))
p <- p + xlab("Fever") + ylab("Log Rel. Abund.") + ggtitle("Waltera")
p <- p + theme(legend.position="none")
p
#dev.off()

#summarize by diet
p <- ggplot(pdata2, aes(x=Diet2, y=log(Acetatifactor), colour=Diet2)) + geom_boxplot(outlier.size=NA, outlier.shape=NA)
p <- p + geom_jitter(width=0.1, height=0)
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + scale_colour_manual(values=c("darkcyan", "lightsalmon3"))
p <- p + xlab("Diet") + ylab("Log Rel. Abund.") + ggtitle("Waltera")
p <- p + theme(legend.position="none")
p

#summarize by diet (3)
pdata2$Diet3 <- factor(pdata2$Diet3, levels=c("Veg", "Omni"))

p <- ggplot(pdata2, aes(x=Diet3, y=log(Acetatifactor), colour=Diet3)) + 
  geom_boxplot(outlier.size=NA, outlier.shape=NA) + 
  geom_jitter(width=0.1, height=0) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  scale_colour_manual(values=c("darkcyan", "lightsalmon3")) + 
  xlab("Diet") + ylab("Log Rel. Abund.") + ggtitle("Waltera") + 
  theme(legend.position="none")
p

#summarize by both diet and fever #still higher by fever within each diet category
#summarize by both diet and fever #still higher by fever within each diet category #Diet3
#cairo_pdf(width=4, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/4G.WalteraDiet3Fever.pdf")
p <- ggplot(pdata2, aes(x=Diet3, y=log(Acetatifactor), colour=fever7d_aboveavg)) + geom_boxplot(outlier.size=NA, outlier.shape=NA) +
  geom_point(position=position_jitterdodge(jitter.width=0.1, jitter.height=0)) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) +
  scale_colour_manual(values=c("grey50", "firebrick3")) + 
  xlab("Diet") + 
  ylab("Log Rel. Abund.") + 
  ggtitle("Waltera") + labs(colour="âˆ†T") +
  annotate('segment', x=0.75, xend=1.25, y=-2.8, yend=-2.8) +
  annotate('text', label=c("p=0.05"), x=1, y=-2.5) +
  annotate('segment', x=1.75, xend=2.25, y=-2.8, yend=-2.8) +
  annotate('text', label=c("*"), x=2, y=-2.5)
p
#dev.off()

wilcox.test(pdata2$Acetatifactor~pdata2$Diet3) #p=0.1
pdata2_veg <- pdata2 %>% filter(Diet3=="Veg")
pdata2_omni <- pdata2 %>% filter(Diet3=="Omni")
wilcox.test(pdata2_veg$Acetatifactor~pdata2_veg$fever7d_aboveavg) #p=0.05
wilcox.test(pdata2_omni$Acetatifactor~pdata2_omni$fever7d_aboveavg) #p=0.04

#over time vs IgG
pdata1$IgG_B2_aboveavg <- factor(pdata1$IgG_B2_aboveavg, levels=c("lo", "hi"))
#pdf(width=4, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/4H.MetagenomicAcetatifactor_IgGB2.pdf")
p <- ggplot(pdata1, aes(x=as.factor(sample_no3), y=log(Acetatifactor), colour=IgG_B2_aboveavg)) + geom_boxplot(outlier.size=NA, outlier.shape=NA) + 
  geom_point(position=position_jitterdodge(jitter.width=0.1, jitter.height=0)) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  scale_colour_manual(values=c("skyblue", "skyblue4")) + 
  annotate("segment", x=3.5, xend=3.5, y=-9, yend=-2, linetype=2) + 
  annotate("text", x=3.5, y=-1.5, label=c("Vaccine")) + 
  xlab("Timepoint") + ylab("Log Rel. Abund.") + ggtitle("Waltera") + labs(colour="IgG")
p
#dev.off()

#longitudinal significance via mixed linear model
m2 <- lme(Acetatifactor ~ seq_depth + Bristol_Stool_Score2 + Time_at_RT + IgG_B2_aboveavg, random=~1|Participant_ID, 
            data = pdata1, 
            na.action=na.exclude, method="ML") #NS
summary(m2) #igg2, p=0.0898, slightly different from Maaslin2 result but basically the same idea

#fold change IgG
pdata1$IgG_riseB1B2
m3 <- lme(Acetatifactor ~ seq_depth + Bristol_Stool_Score2 + Time_at_RT + IgG_riseB1B2, random=~1|Participant_ID, 
            data = pdata1, 
            na.action=na.exclude, method="ML") #NS
summary(m3) #p=0.6676

pdata1$IgG_foldB1B2
pdata2 <- pdata1 %>% filter(IgG_foldB1B2!="Inf")
m3 <- lme(Acetatifactor ~ seq_depth + Bristol_Stool_Score2 + Time_at_RT + IgG_foldB1B2, random=~1|Participant_ID, 
            data = pdata2, 
            na.action=na.exclude, method="ML") #NS
summary(m3) #p=0.6239



#what about Waltera abundance vs. MEDI reads?
#export a Waltera rel abund table to add to the MEDI analysis.

waltera <- pdata1 %>% dplyr::select(Acetatifactor, SampleID, Participant_ID)
#write.csv(waltera, "/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_input/waltera_abundance.csv")

```

```{r genes dna maaslin2}
#gene-level analysis
hm3_genes_dna <- fread("/ebio/abt3_projects2/uHEAT/data/output_LLMGP_humann_all_rep/output_LLMGP_humann_pool3/humann3/normalized/unstratified/regroup/annot/genefamilies_uniref50_default.tsv")
names(hm3_genes_dna) <- gsub("_Abundance-RPKs", "", names(hm3_genes_dna), fixed=TRUE)
head(hm3_genes_dna)
hm3_genes_dna <- hm3_genes_dna %>% rename(Gene=`# Gene Family`)

hm3_genes_dna_M <- hm3_genes_dna %>% select(-c("annotation", "Gene"))
head(hm3_genes_dna_M)
hm3_genes_dna_M <- as.data.frame(hm3_genes_dna_M)
row.names(hm3_genes_dna_M) <- hm3_genes_dna$Gene

meta_seq <- meta_seq[order(meta_seq$SampleID),] #order
hm3_genes_dna_M <- hm3_genes_dna_M[,which(names(hm3_genes_dna_M) %in% meta_seq$SampleID)] 
hm3_genes_dna_M <- hm3_genes_dna_M[,order(names(hm3_genes_dna_M))] #order
names(hm3_genes_dna_M)==meta_seq$SampleID #cleaned up again
row.names(meta_seq) <- meta_seq$SampleID
row.names(hm3_genes_dna_M)

hm3_genes_dna <- NULL #because it's so big, get rid of it!

#test by gene - this takes hours
fit_data = Maaslin2(
  input_data = hm3_genes_dna_M, 
  input_metadata = meta_seq, 
  output = "maaslin2_output", 
  fixed_effects = c("seq_depth", "Bristol_Stool_Score", "Time_at_RT", "fever7d_aboveavg"), 
  random_effects = c("Participant_ID"), 
  #reference=c("Sex", "m"), #tell it to use plate 1 as the ref for plate compare / male as ref for Sex
  normalization="tss", #total sum scaling
  min_abundance=0.00001, #expand the threshold from 10^-3 to 10%-4 because genes?
  min_prevalence=0.25, #must be present in 25% of samples
  max_significance=0.05, #calm down with the non significant associations and plots
  plot_scatter=FALSE
)
res_long_fevergenes_dna <- fit_data$results 
res_long_fevergenes_dna %>% filter(qval<0.05&metadata=="fever_aboveavg") 

#export immediately
hits_dna <- fit_data$results %>% filter(pval<0.05&metadata=="fever_aboveavg")
hits_dna
genes_dna <- gene_key %>% dplyr::rename(feature=Gene) %>% 
  full_join(hits_dna) %>% filter(!is.na(coef))
#write.csv(genes_dna, "/ebio/abt3_projects2/uHEAT2/data/output_LLMGP_RNA_all/maaslin2_output/Uniref50_fever_original_DNA_hits.csv")
genes_dna[grep("flagell*", genes_dna$annotation, ignore.case=TRUE),] #FlgB comes up, not sig though; FliC does not
genes_dna[grep("flagell*", genes_dna$feature, ignore.case=TRUE),] #FlgB comes up, not sig though; FliC does not

```


```{r volcano time interaction3 }
#plot hits as a volcano summary
#interaction effect of fever7d and stooltiming3
time3_fever_int <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/Rscripts_server/maaslin2_temporal_output/DNAg_seqBSSTimeRT_FeverTime3_Interaction.csv", row.names=1)

vdata <- time3_fever_int %>% filter(metadata=="FeverTime")

mycolours <- c("deepskyblue4", "deepskyblue3", "firebrick4", "firebrick2", "grey")
names(mycolours) <- c("DOWN q<0.05", "DOWN q<0.1", "UP q<0.05", "UP q<0.1",  "NO")
vdata$diffexpressed <- c("NO")
#vdata$diffexpressed[vdata$coef > 0 & vdata$qval<0.1] <- c("UP q<0.1")
vdata$diffexpressed[vdata$coef > 0 & vdata$qval<0.05] <- c("UP q<0.05")
#vdata$diffexpressed[vdata$coef < 0 & vdata$qval<0.1] <- c("DOWN q<0.1")
vdata$diffexpressed[vdata$coef < 0 & vdata$qval<0.05] <- c("DOWN q<0.05")

# Create a new column "delabel" to de, that will contain the name of genes differentially expressed 
vdata$delabel <- ifelse(vdata$diffexpressed!="NO",
                          yes=vdata$feature, no=NA)

#cairo_pdf(width=5, height=3.5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S2F.DNA.TimeFever.pdf")
p = ggplot(data=vdata, aes(x=coef, y=-log10(pval), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle(expression("DNA âˆ†T"^hi~"After Vaccine")) +
  scale_colour_manual(values = mycolours) + 
  geom_point(size=0.3) + 
  theme_bw() +
  guides(fill = guide_legend(override.aes = aes(label = ""))) + 
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE)
#    geom_text(size=2)
p
#dev.off()

#make a legend

#export the significant hits for Table S#.
vdata_export <- vdata %>% filter(qval<0.1) %>% dplyr::select(feature, value, coef, pval, qval, N, N.not.zero)
vdata_export$Covariates <- c("seq_depth+BSS+Time_at_RT+fever+Time")
#write.csv(vdata_export, "/ebio/abt3_projects2/uHEAT/data/SupplementalTables/Table#_FigS2F.csv")


```

```{r volcano time fever hi only}
#plot hits as a volcano summary
#time3 (before/after) in fever only
time3_fever_hi <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_temporal_output/DNAg_seqBSSTimeRT_Time3_hi.csv", row.names=1)

vdata <- time3_fever_hi %>% filter(metadata=="Stool_timing3")

mycolours <- c("deepskyblue4", "deepskyblue3", "firebrick4", "firebrick2", "grey")
names(mycolours) <- c("DOWN q<0.05", "DOWN q<0.1", "UP q<0.05", "UP q<0.1",  "NO")
vdata$diffexpressed <- c("NO")
vdata$diffexpressed[vdata$coef > 0 & vdata$qval<0.1] <- c("UP q<0.1")
vdata$diffexpressed[vdata$coef > 0 & vdata$qval<0.05] <- c("UP q<0.05")
vdata$diffexpressed[vdata$coef < 0 & vdata$qval<0.1] <- c("DOWN q<0.1")
vdata$diffexpressed[vdata$coef < 0 & vdata$qval<0.05] <- c("DOWN q<0.05")

# Create a new column "delabel" to de, that will contain the name of genes differentially expressed 
vdata$delabel <- ifelse(vdata$diffexpressed!="NO",
                          yes=vdata$feature, no=NA)

#pdf(width=5, height=3.5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S2F.DNA.TimeFever.pdf")
p = ggplot(data=vdata, aes(x=coef, y=-log10(pval), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle("DNA Fever-Hi: After vs Before") +
  scale_colour_manual(values = mycolours) + 
  geom_point(size=0.3) + 
  theme_bw() +
  guides(fill = guide_legend(override.aes = aes(label = ""))) + 
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE)
#    geom_text(size=2)
p
#dev.off()

```

```{r volcano time fever lo only}
#plot hits as a volcano summary
#time3 (before/after) in fever only
time3_fever_lo <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_temporal_output/DNAg_seqBSSTimeRT_Time3_lo.csv", row.names=1)

vdata <- time3_fever_lo %>% filter(metadata=="Stool_timing3")

mycolours <- c("deepskyblue4", "deepskyblue3", "firebrick4", "firebrick2", "grey")
names(mycolours) <- c("DOWN q<0.05", "DOWN q<0.1", "UP q<0.05", "UP q<0.1",  "NO")
vdata$diffexpressed <- c("NO")
vdata$diffexpressed[vdata$coef > 0 & vdata$qval<0.1] <- c("UP q<0.1")
vdata$diffexpressed[vdata$coef > 0 & vdata$qval<0.05] <- c("UP q<0.05")
vdata$diffexpressed[vdata$coef < 0 & vdata$qval<0.1] <- c("DOWN q<0.1")
vdata$diffexpressed[vdata$coef < 0 & vdata$qval<0.05] <- c("DOWN q<0.05")

# Create a new column "delabel" to de, that will contain the name of genes differentially expressed 
vdata$delabel <- ifelse(vdata$diffexpressed!="NO",
                          yes=vdata$feature, no=NA)

#pdf(width=5, height=3.5, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S2F.DNA.TimeFever.pdf")
p = ggplot(data=vdata, aes(x=coef, y=-log10(pval), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle("DNA Fever-Lo: After vs Before") +
  scale_colour_manual(values = mycolours) + 
  geom_point(size=0.3) + 
  theme_bw() +
  guides(fill = guide_legend(override.aes = aes(label = ""))) + 
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE)
#    geom_text(size=2)
p
#dev.off()

```


