---
title: "08.Waltera"
author: "Kelsey Huus"
date: "2024-08-02"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(ggplot2)
library(tidyr)
library(edgeR)
library(ggplot2)
library(biomaRt)
#library(data.table)
library(ggrepel)

```

```{r cytokines plot}
il8_4h_all <- read.csv("/ebio/abt3_projects2/uHEAT/data/Organoid_RNAseq/cytokines/IL8_4h_bothreps.csv", row.names=1)

pdata <- il8_4h_all %>% filter(FeverFlagellin %in% c("Hi", "Lo"))
pdata$Fever <- factor(pdata$FeverFlagellin, levels=c("Lo", "Hi"))

wilcox.test(pdata$IL8~pdata$FeverFlagellin) #p=0.040
#double check final N
dim(pdata)
pdata$Fever <- gsub("Lo", "lo", pdata$Fever)
pdata$Fever <- gsub("Hi", "hi", pdata$Fever)
pdata$Fever <- factor(pdata$Fever, levels=c("lo", "hi"))

#cairo_pdf(width=3, height=3, "//ebio/abt3_projects2/uHEAT/data/CleanPlots/5A.IL8.4h.pdf")
p <- ggplot(pdata, aes(x=Fever, y=IL8)) + geom_boxplot(outlier.shape=NA, outlier.size=NA) + 
  geom_jitter(width=0.1, height=0, size=3) + 
  ylab("IL8 (pg/mL)") + 
  annotate('segment', x=1, xend=2, y=2800, yend=2800) + 
  annotate('text', x=1.5, y=2875, label=c("*"), size=5) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  ylab("IL-8 (pg/mL)") + xlab("∆T")
p
#dev.off()

#pdf(width=4, height=4, "/Users/khuus/ownCloud/KH-Postdoc/µHeat/OrganoidExperiments/4h_IL8_bothreps_byrep.pdf")
p <- ggplot(pdata, aes(x=Fever, y=IL8, shape=Rep)) + geom_boxplot(outlier.shape=NA, outlier.size=NA) + geom_point(position=position_jitterdodge(jitter.width=0.1, jitter.height=0))
p <- p + ylab("IL8 (pg/mL)")
p
#dev.off()

il8_18h_all <- read.csv("/ebio/abt3_projects2/uHEAT/data/Organoid_RNAseq/cytokines/IL8_18h_bothreps.csv", row.names=1)

pdata <- il8_18h_all %>% filter(FeverFlagellin %in% c("Hi", "Lo"))
pdata$Fever <- factor(pdata$FeverFlagellin, levels=c("Lo", "Hi"))
wilcox.test(pdata$IL8~pdata$FeverFlagellin) #p=0.044 
pdata$Fever <- gsub("Lo", "lo", pdata$Fever)
pdata$Fever <- gsub("Hi", "hi", pdata$Fever)
pdata$Fever <- factor(pdata$Fever, levels=c("lo", "hi"))

#cairo_pdf(width=3, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/5A.IL8.18h.pdf")
p <- ggplot(pdata, aes(x=Fever, y=IL8)) + geom_boxplot(outlier.shape=NA, outlier.size=NA) + geom_jitter(width=0.1, alpha=0.8, height=0) + 
  ylab("IL8 (pg/mL)") + 
  annotate('segment', x=1, xend=2, y=12500, yend=12500) + 
  annotate('text', x=1.5, y=12650, label=c("*")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) + 
  ylab("IL-8 (pg/mL)") + xlab("∆T")
p
#dev.off()

#pdf(width=4, height=4, "/Users/khuus/ownCloud/KH-Postdoc/µHeat/OrganoidExperiments/18h_IL8_bothreps_byrep.pdf")
p <- ggplot(pdata, aes(x=Fever, y=IL8, shape=Rep)) + geom_boxplot(outlier.shape=NA, outlier.size=NA) + geom_point(position=position_jitterdodge(jitter.width=0.1, jitter.height=0))
p <- p + ylab("IL8 (pg/mL)")
p
#dev.off()

#plot both timepoints together.
il8_4h_all$Time <- c("4h")
il8_18h_all$Time <- c("18h")
pdata <- full_join(il8_4h_all, il8_18h_all)
pdata <- pdata %>% filter(FeverFlagellin %in% c("Hi", "Lo"))
pdata$Fever <- factor(pdata$FeverFlagellin, levels=c("Lo", "Hi"))
pdata$Time <- factor(pdata$Time, levels=c("4h", "18h"))

#pdf(width=4.5, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/4A.IL8.4h18h.pdf")
p <- ggplot(pdata, aes(x=Fever, y=IL8)) + geom_boxplot(outlier.shape=NA, outlier.size=NA) + geom_jitter(width=0.1, alpha=0.8, height=0, size=3)
p <- p + ylab("IL8 (pg/mL)")
p <- p + annotate('segment', x=1, xend=2, y=12500, yend=12500)
p <- p + annotate('text', x=1.5, y=12650, label=c("*"))
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + ylab("IL-8 (pg/mL)")
p <- p + facet_grid(.~Time)
p
#dev.off()

#plot controls at both timepoints for supplemental
pdata1 <- full_join(il8_4h_all, il8_18h_all)
pdata1 <- pdata1 %>% filter(FeverFlagellin %in% c("Blank", "FliC"))
pdata1$Time <- factor(pdata1$Time, levels=c("4h", "18h"))

p <- ggplot(pdata1, aes(x=FeverFlagellin, y=IL8)) + geom_boxplot(outlier.shape=NA, outlier.size=NA) + geom_jitter(width=0.1, alpha=0.8, height=0, size=3)
p <- p + ylab("IL8 (pg/mL)")
#p <- p + annotate('segment', x=1, xend=2, y=12500, yend=12500)
#p <- p + annotate('text', x=1.5, y=12650, label=c("*"))
p <- p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10)))
p <- p + ylab("IL-8 (pg/mL)") + xlab("Control")
p <- p + facet_grid(.~Time)
p

#re-export the organoid data ID's to combine in a supplemental table.
orga_export <- il8_4h_all %>% dplyr::select(SampleID, IL8)
orga_export$Participant_ID <- gsub("H", "HEAT_", orga_export$SampleID)
#write.csv(orga_export, "/ebio/abt3_projects2/uHEAT/data/SupplementalTables/S#_cellculture.csv")

```

```{r rnaseq import}
#Process the samples similarly to Alex using his script (edgeR) - or myself DESeq2?
work.dir = "/ebio/abt3_projects2/uHEAT/data/Organoid_RNAseq"
out.dir = "/ebio/abt3_projects2/uHEAT/data/Organoid_RNAseq/edgeR"
#collect files
# Reading files
files = list.files(path = file.path(work.dir, "star/STAR_out"),
                   pattern = "ReadsPerGene.out.tab", recursive = T, include.dirs = T, full.names = T)
files
sample_names <- sapply(strsplit(files,"STAR_out/"), `[`, 2)
sample_names <- gsub("ReadsPerGene.out.tab", "", sample_names)
#add Participant ID and sample ID / rep info
samples <- data.frame(Sample=sample_names)
samples$Participant_ID <- sapply(strsplit(samples$Sample,"_"), `[`, 1)
samples$Participant_ID <- gsub("H", "HEAT_", samples$Participant_ID)
samples$Technical_rep <- sapply(strsplit(samples$Sample,"_"), `[`, 2)
head(samples)

#df <- data.frame(File=files, SampleID=sample_names)
#head(df)

#can add metadata here.
meta <- read.csv("/ebio/abt3_projects2/uHEAT/data/metadata/metadata_organoids_rep1rep2_combined.csv")
meta <- meta %>% dplyr::select(-c(Sample, SampleID.x, SampleID.y, X.1, X)) %>% 
  filter(!duplicated(Participant_ID)) %>%
  full_join(samples, by="Participant_ID") %>% 
  filter(!is.na(Sample)&!duplicated(Sample))
meta %>% dplyr::select(Sample, Pair, fever7d_aboveavg)
meta$FeverFlagellin <- meta$fever7d_aboveavg
meta$Ctrls <- ifelse(is.na(meta$fever7d_aboveavg), yes=meta$Sample, no=NA)
meta$Ctrls <- sapply(strsplit(meta$Ctrls,"_"), `[`, 2)
meta$Ctrls <- ifelse(meta$Ctrls %in% c("p88", "PBS"), 
                              yes="Neg_ctrl", no=meta$Ctrls)
meta$FeverFlagellin <- ifelse(is.na(meta$FeverFlagellin), yes=meta$Ctrls, no=meta$FeverFlagellin)
meta$FeverFlagellin
meta <- meta[order(meta$Sample),]
meta$SampleID <- sapply(strsplit(meta$Sample,"_"), `[`, 1)
#meta$SampleID <- ifelse(meta$SampleID %in% c("Rep1", "Rep2", "REP2"),
#                        yes=sapply(strsplit(meta$Sample,"_"), `[`, 2),
#                                   no=meta$SampleID)
meta$SampleID <- ifelse(meta$SampleID %in% c("Rep1", "Rep2", "REP2"),
                        yes=meta$Sample, no=meta$SampleID) #to keep each FliC rep separate 

#function for reading and merging text files containing gene expression counts
DGE.obj <- readDGE(files, labels=sample_names)
DGE.obj$samples
DGE.obj$counts

```

```{r rnaseq qc and clean}

# Delete unnecessary read count rows
hk <- DGE.obj$counts[!(rownames(DGE.obj$counts) %in% c('N_multimapping', 'N_noFeature',
                                                      'N_ambiguous', 'N_unmapped')),]
#and re-calculate lib sizes
DGE.obj$samples$lib.size <- colSums(hk)
DGE.obj$counts <- DGE.obj$counts[!(rownames(DGE.obj$counts) %in% c('N_multimapping', 'N_noFeature',
                                                       'N_ambiguous', 'N_unmapped')),]

###Summarize by technical replicates.
identical(colnames(DGE.obj), meta$Sample) #TRUE
matrix.tmp <- DGE.obj$counts
colnames(matrix.tmp) = meta$SampleID
matrix.sum <- sumTechReps(matrix.tmp, ID=colnames(matrix.tmp)) 
DGE.obj.sum <- DGEList(matrix.sum) #oh and then I can reconvert it, okay cool.
DGE.obj.sum

# Libraries sizes plot
par(mar = c(8, 5, 2, 0))
barplot(DGE.obj$samples$lib.size*1e-6, names=rownames(DGE.obj$samples), 
        ylab="Library size (millions)", 
        #xlab="Samples",
        las=2)

barplot(DGE.obj.sum$samples$lib.size*1e-6, names=rownames(DGE.obj.sum$samples), 
        ylab="Library size (millions)", 
        #xlab="Samples",
        las=2)

#need to define the design here in order to filter correctly.
#add metadata for grouping.
meta_notech <- meta %>% filter(Technical_rep!=2)
#double check, I am not convinced this is joining correctly! #though inner join should match
identical(colnames(DGE.obj.sum), meta_notech$SampleID) 
#instead of excluding groups here, exclude them later. merge the fever info
DGE.obj.sum$group = tibble(Sample = samples$Sample) %>% 
  inner_join(meta_notech) %>% 
  pull(FeverFlagellin) 
check <- data.frame(Sample=row.names(DGE.obj.sum$samples), Fever=DGE.obj.sum$group)
identical(check$Sample, meta_notech$SampleID) #seems fine
identical(check$Fever, meta_notech$FeverFlagellin) #seems fine

##*Define the design of the experiment## #note - this affects filtering
#start with the controls only as a test case.
group <- DGE.obj.sum$group
group <- factor(group)
str(group)
data.frame(row.names(DGE.obj.sum$samples), group) #check
DGE.obj.sum$samples$group <- DGE.obj.sum$samplesgroup_ctrls

my_design <- model.matrix(~0+group, data=DGE.obj.sum$samples)
colnames(my_design) <- levels(group)
my_design

#Filter: discard poorly-expressed genes
keep <- filterByExpr(DGE.obj.sum, design = my_design)
table(keep)

DGE.obj.k <- DGE.obj.sum[keep, keep.lib.sizes=FALSE]

# Normalization w/factor calculation 
DGE.obj.n <- calcNormFactors(DGE.obj.k, method="TMM")
DGE.obj.n

```

```{r rnaseq annotate}
#add gene annotations from human genome #to the normalized/filtered table.
geneid <- rownames(DGE.obj.n$counts)
length(geneid)
mart38 <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl")
mart38
f <- listFilters(mart38)
atr <- listAttributes(mart38)
resBM <- biomaRt::getBM(attributes = c('ensembl_gene_id', 
                                       'external_gene_name',
                                       'entrezgene_id',
                                       'description', 'gene_biotype'),
                        filters = 'ensembl_gene_id', 
                        values = geneid,
                        mart = mart38,
                        useCache = FALSE)


# find ambiguous genes (duplicated annotations)
resgenes <- as.data.table(resBM)
non.unique <- resgenes[, .N, by=entrezgene_id][N>=2]$entrezgene_id
non.unique.2 <- resgenes[, .N, by=ensembl_gene_id][N>=2]$ensembl_gene_id
# to be removed
non.unique
non.unique.2
# remove them.
ids.good <- resgenes[!is.na(resgenes$entrezgene_id) & 
                  !(entrezgene_id %in% non.unique) & 
                  !(ensembl_gene_id %in% non.unique.2)]$ensembl_gene_id
genes <- resgenes[ensembl_gene_id %in% ids.good,]$entrezgene_id
names(genes) <- resgenes[ensembl_gene_id %in% ids.good,]$ensembl_gene_id
length(genes)

#make an annotated DGE object
DGE.obj.entrez <- copy(DGE.obj.n) #Alex did it from the normalized table
DGE.obj.entrez$counts <- DGE.obj.entrez$counts[rownames(DGE.obj.entrez$counts) %in% names(genes), ]
#Change ensembl names to entrez names
rownames(DGE.obj.entrez$counts) <- genes[rownames(DGE.obj.entrez$counts)]
DGE.obj.entrez$counts

entrez.descr <- resgenes[, c("external_gene_name", "entrezgene_id", "description", "gene_biotype", "ensembl_gene_id")]
entrez.descr$entrezgene_id <- as.character(entrez.descr$entrezgene_id)
entrez.descr
entrez.descr %>% filter(external_gene_name == "TLR5") #for example.

```

```{r rnaseq test}
#Prepare design matrix for testing
DGE.obj.entrez$group
contr.matrix <- makeContrasts(
  hi_vs_lo = hi - lo,
  p15_vs_ctrl = p15 - Neg_ctrl,
  FliC_vs_ctrl = FliC - Neg_ctrl,
  FliC_vs_p15 = FliC - p15,
  levels = colnames(my_design))
contr.matrix

#make logCPM to export for the whole table.
y = DGE.obj.entrez
logCPM <- cpm(y, prior.count=1, log=TRUE) #cpm with log2

# rename the genes from Ensembl IDs to normal names (external_gene_name)
ggg <- resgenes[ensembl_gene_id %in% ids.good,]$external_gene_name
names(ggg) <- resgenes[ensembl_gene_id %in% ids.good,]$entrezgene_id
rownames(logCPM) <- ggg[rownames(logCPM)]
logCPM

#write.table(logCPM, file.path(out.dir, "all_logCPM.txt"), 
#            row.names = T, quote = F, sep = '\t')

#test ctrls
curr_contrast <- c("FliC_vs_ctrl") #start with control

#Calculate Dispersion (common and by genes)
ydisp <- estimateDisp(y, my_design, robust=TRUE)
plotBCV(ydisp)   
#Fit GLM
fit <- glmFit(ydisp, my_design)

#Make a likelihood ratio test
lrt <- glmLRT(fit, contrast = contr.matrix[, curr_contrast])
topTags(lrt)
summary(decideTests(lrt, adjust.method = 'fdr', p.value=0.05)) #ah. that makes more sense.

#individual counts-per-million for the top genes
res_flic <- as.data.frame(lrt$table[order(lrt$table$PValue),])
res_flic$FDR <- p.adjust(res_flic$PValue, 'fdr')
res_flic$entrezgene_id <- row.names(res_flic) #the row names are entrez gene ids
# Add gene annotation and make a table for GSEA
res_flic <- res_flic %>% full_join(entrez.descr)
logFC_flic <- data.frame(ids=res_flic$external_gene_name, scores=res_flic$logFC) #keep all results for GSEA analysis
logFC_flic <- logFC_flic[order(logFC_flic$scores),]
names(logFC_flic) <- NULL
head(logFC_flic)
res_flic_sig <- res_flic %>% filter(FDR<0.05)
res_flic_sig
#export
#write.table(res_flic_sig, file.path(out.dir, paste0(curr_contrast, '_hits.txt')), 
#            row.names = F, quote = F, sep = '\t')

#repeat here for fever hi vs fever lo. :)
curr_contrast <- c("hi_vs_lo")

#Make a likelihood ratio test
lrt <- glmLRT(fit, contrast = contr.matrix[, curr_contrast])
topTags(lrt)
summary(decideTests(lrt, adjust.method = 'fdr', p.value=0.05)) 

#individual counts-per-million for the top genes
res_fever <- as.data.frame(lrt$table[order(lrt$table$PValue),])
res_fever$FDR <- p.adjust(res_fever$PValue, 'fdr')
res_fever$entrezgene_id <- row.names(res_fever) #the row names are entrez gene ids
# Add gene annotation and make a FC table for GSEA
res_fever <- res_fever %>% full_join(entrez.descr)
logFC_hilo <- data.frame(ids=res_fever$external_gene_name, scores=res_fever$logFC) #keep all results for GSEA analysis
logFC_hilo <- logFC_hilo[order(logFC_hilo$scores),]
names(logFC_hilo) <- NULL
head(logFC_hilo)
#and export significant results
res_fever_sig <- res_fever %>% filter(PValue<0.05) #take the raw values just to check.
res_fever_sig
#write.table(res_fever_sig, file.path(out.dir, paste0(curr_contrast, '_hits.txt')), 
#            row.names = F, quote = F, sep = '\t')

#what is the overlap between flagellin hits and fever hits?
FliC_vs_Ctrl_up <- res_flic_sig %>% filter(logFC>0) #qval corrected
hi_vs_lo_up <- res_fever_sig %>% filter(logFC>0) #only pval corrected
length(FliC_vs_Ctrl_up$external_gene_name) #223 hits
length(hi_vs_lo_up$external_gene_name) #139 hits that don't pass FDR
flagellin_fever_genes <- intersect(FliC_vs_Ctrl_up$external_gene_name, hi_vs_lo_up$external_gene_name)
flagellin_fever_genes <- hi_vs_lo_up %>% filter(external_gene_name %in% flagellin_fever_genes) %>% 
  dplyr::select(c(external_gene_name, description))
flagellin_fever_genes #includes NFKB and plenty of classic inflammation genes; and IL32
#write.table(flagellin_fever_genes, paste0(out.dir, '/flagellin_fever_genes_overlap.txt'),
#            row.names = F, quote = F, sep = '\t')

dim(flagellin_fever_genes) #78 overlap
length(flagellin_fever_genes$external_gene_name)/length(hi_vs_lo_up$external_gene_name)*100 #so more than half overlap with FliC FDR correction

```

```{r rnaseq volcano plots}
#make a volcano plot of FliC vs ctrl
mycolours <- c("blue", "red", "grey")
names(mycolors) <- c("DOWN", "UP", "NO")
res_flic$diffexpressed <- c("NO")
res_flic$diffexpressed[res_flic$logFC > 0 & res_flic$FDR<0.05] <- c("UP")
res_flic$diffexpressed[res_flic$logFC < 0 & res_flic$FDR<0.05] <- c("DOWN")

# Now write down the name of genes beside the points...
# Create a new column "delabel" to de, that will contain the name of genes differentially expressed (N
res_flic$delabel <- ifelse(res_flic$diffexpressed!="NO",
                          yes=res_flic$external_gene_name, no=NA)

#pdf(width=8, height=6, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S4A.FliCVolcano.pdf")
p = ggplot(data=res_flic, aes(x=logFC, y=-log10(PValue), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle("FliC vs Ctrl") +
  scale_colour_manual(values = mycolors) + 
  geom_point(size=0.3) + 
  theme_bw() +
  guides(fill = guide_legend(override.aes = aes(label = ""))) + 
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE)
#    geom_text(size=2)
p
#dev.off()

#repeat for hi vs lo - label the top ones
res_fever$diffexpressed <- c("NO")
res_fever$diffexpressed[res_fever$logFC > 1 ] <- c("UP")
res_fever$diffexpressed[res_fever$logFC < -1 ] <- c("DOWN")
res_fever$delabel <- ifelse(res_fever$diffexpressed!="NO",
                           yes=res_fever$external_gene_name, no=NA)
res_fever$diffexpressed <- c("NO")

#pdf(width=8, height=6, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/S4B.FeverVolcano.pdf")
p = ggplot(data=res_fever, aes(x=logFC, y=-log10(PValue), 
                              col=diffexpressed, 
                              label=delabel)) + 
  ggtitle("Fever Hi vs Lo") +
  scale_colour_manual(values = 'black') + 
  geom_point(size=0.3) + 
  theme_bw() +
  guides(fill = guide_legend(override.aes = aes(label = ""))) + 
  geom_text_repel(size=2,  max.overlaps = 30, segment.size=0.2, show.legend = FALSE)
#    geom_text(size=2)
p
#dev.off()

```

```{r rnaseq gsea plot}
#website WebGestalt was used on the fold changes list of feverhi vs feverlo

gsea_res <- read.delim("/ebio/abt3_projects2/uHEAT/data/Organoid_RNAseq/gsea/enrichment_results_wg_result1720512139.txt")
gsea_res
gsea_res$Sig <- ifelse(gsea_res$FDR<0.05, "yes", "no")

pdata <- gsea_res %>% filter(Sig=="yes")
pdata$description <- factor(pdata$description, levels=c(pdata[order(pdata$normalizedEnrichmentScore),]$description))

#summary plot of gsea results
#pdf(width=6, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/6B.GSEA.pdf")
p <- ggplot(pdata, aes(x=normalizedEnrichmentScore, y=description, fill=FDR)) + geom_bar(stat='identity', colour='black')
p <- p + scale_fill_gradient(low="aliceblue", high="deepskyblue3")
p <- p + theme_bw() + ylab(NULL) + labs(fill="qval") + ggtitle("RNAseq: GSEA")
p
#dev.off()

#reexport the sig hits for a supplemental table
#write.csv(pdata, "/ebio/abt3_projects2/uHEAT/data/SupplementalTables/TableS#_Fig5B.csv")

```

```{r pbmcs}
ldh <- read.csv('/ebio/abt3_projects2/uHEAT/data/PBMCs/24.10.17.PBMC.FecalSups.LDHRelease.ForR.csv') #cell death
IL18 <- read.csv("/ebio/abt3_projects2/uHEAT/data/PBMCs/24.10.18.PBMCs.IL18Production.csv")
IL1B <- read.csv("/ebio/abt3_projects2/uHEAT/data/PBMCs/24.10.21.PBMCs.IL1BProduction.csv")
meta1 <- read.csv("/ebio/abt3_projects2/uHEAT/data/PBMCs/Organoid_Samples_metadata.csv")
meta2 <- read.csv("/ebio/abt3_projects2/uHEAT/data/PBMCs//Organoid_Samples_metadata_rep2.csv")

meta <- full_join(meta1, meta2)
meta$fever7d_aboveavg <- factor(meta$fever7d_aboveavg, levels=c("lo", "hi"))

#Cell death - LDH
ldh_meta <- full_join(ldh, meta)
head(ldh_meta)

pdata <- ldh_meta %>% filter(!is.na(fever7d_aboveavg)) #exclude the flagellin/pbs controls


#cairo_pdf(width=2.5, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/5E_PBMC_FecalSups_LDH.pdf")
p <- ggplot(pdata, aes(x=fever7d_aboveavg, y=LDH_60min)) + 
  geom_boxplot(outlier.shape=NA) + geom_jitter(width=0.1, height=0, size=3) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) +
  annotate('segment', x=1, xend=2, y=0.83, yend=0.83) + 
  annotate('text', x=1.5, y=0.86, label='*', size=5) + 
  ylab("LDH Release") + xlab("∆T")
p
#dev.off()

wilcox.test(pdata$LDH_60min~pdata$fever7d_aboveavg) #p=0.03

IL18_meta <- IL18 %>%
  dplyr::rename(Participant_ID=SampleID) %>%
  full_join(meta)

pdata <- IL18_meta %>% filter(!is.na(fever7d_aboveavg))  #exclude the flagellin/pbs controls
pdata$fever7d_aboveavg <- factor(pdata$fever7d_aboveavg, levels=c("lo", "hi"))

#cairo_pdf(width=2.5, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/5C_PBMC_FecalSups_IL18.pdf")
p <- ggplot(pdata, aes(x=fever7d_aboveavg, y=IL18)) + 
  geom_boxplot(outlier.shape=NA) + geom_jitter(width=0.1, height=0, size=3) +
 theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) +
  annotate('segment', x=1, xend=2, y=206, yend=206) + 
  annotate('text', x=1.5, y=210, label='*', size=5) + 
  ylab("IL-18 (pg/mL)") + xlab("∆T")
p
#dev.off()
wilcox.test(pdata$IL18~pdata$fever7d_aboveavg) #p=0.044

#IL1B
IL1B$IL1B <- IL1B$IL1B_1.10*10
IL1B_meta <- IL1B %>%
  dplyr::rename(Participant_ID=Sample) %>%
  full_join(meta)

pdata <- IL1B_meta %>% filter(!is.na(fever7d_aboveavg))  #exclude the flagellin/pbs controls
pdata$fever7d_aboveavg <- factor(pdata$fever7d_aboveavg, levels=c("lo", "hi"))

#cairo_pdf(width=2.5, height=3, "/ebio/abt3_projects2/uHEAT/data/CleanPlots/5D_PBMC_FecalSups_IL1B.pdf")
p <- ggplot(pdata, aes(x=fever7d_aboveavg, y=IL1B)) + 
  geom_boxplot(outlier.shape=NA) + geom_jitter(width=0.1, height=0, size=3) +
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.title.y = element_text(margin = margin(r = 10))) +
  annotate('segment', x=1, xend=2, y=5750, yend=5750) + 
  annotate('text', x=1.5, y=5850, label='*', size=5) + 
  ylab("IL-1B (pg/mL)") + xlab("∆T")
p
#dev.off()
wilcox.test(pdata$IL1B~pdata$fever7d_aboveavg) #p=0.023


```

