---
title: "Btheta.Pcopri.RNAseq"
author: "Kelsey Huus"
date: "2024-10-31"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
##Packages
library(dplyr)
library(ggplot2)
#library(data.table)
library(tidyr)

library(DESeq2)
library(stringr)
library(pheatmap)

```

```{r import}

p <- read.delim("/ebio/abt3_projects2/cophylogeny_comparative_genomics/data/rnaseq/llrna/output_llrna_p2/gene_count/counts.txt", 
                sep="\t", header=FALSE)
head(p)
#clean up the weird commenty bit
p <- p[-1,]
names(p) <- p[1,]
p <- p[-1,]
#and clean up the sample names
names(p) <- unlist(lapply(names(p), function(x) str_sub(x, start=-9)))
names(p) <- gsub( ".bam", "", names(p), fixed=TRUE)
#ok?
head(p,3)

#map of metadata
map <- read.delim("/ebio/abt3_projects2/cophylogeny_comparative_genomics/data/rnaseq/llrna/samples_p.txt")
map

```

```{r normalize}
#Prevotella
head(p) #raw counts
p_norm <- p[,7:18] %>% mutate_all(function(x) as.numeric(as.character(x)))  
p_norm <- p_norm/(colSums(p_norm)) #relative abundance per sample
row.names(p_norm) <- p$Geneid

#read in DESEQ2 hits
sig_32p <- read.csv("/ebio/abt3_projects2/cophylogeny_comparative_genomics/data/rnaseq/P_37vs32_deseq2hits_v2_1.5x.csv")
sig_42p <- read.csv("/ebio/abt3_projects2/cophylogeny_comparative_genomics/data/rnaseq/p_42vs37_deseq2hits_v2_1.5x.csv") 

#filter by significance
p_norm_sig <- p_norm[which(row.names(p_norm) %in% (sig_32p$gene) | row.names(p_norm) %in% (sig_42p$gene)), ]
row.names(p_norm_sig) <- gsub("gene-LK433_", "RS", row.names(p_norm_sig))
dim(p_norm_sig) #n=167 (1.5x) or n=58 (2x)

```

```{r heatmap}

#add column and row annotations
#columns
names(p_norm_sig)
my_col_names <- c("32ºC", "32ºC", "32ºC", "32ºC",
                  "37ºC", "37ºC", "37ºC", "37ºC",
                  "42ºC", "42ºC", "42ºC", "42ºC")
my_col_names <- data.frame("Temperature"=my_col_names, row.names=colnames(p_norm_sig))

pheatmap(p_norm_sig,
         scale="row",
         annotation_col = my_col_names,
         annotation_colors = list(Temperature=c("32ºC"="deepskyblue", "37ºC"="darkgrey", "42ºC"="darkorange")),
         main = "P. copri Temperature Shock")


#pdf("/ebio/abt3_projects2/uHEAT/data/CleanPlots/3K_PcopriHeatmap.pdf", width=5, height=5)
pheatmap(p_norm_sig,
         scale="row",
         annotation_col = my_col_names,
         annotation_colors = list(Temperature=c("32ºC"="deepskyblue", "37ºC"="darkgrey", "42ºC"="darkorange")),
         show_rownames=FALSE)
#dev.off()

```
