---
title: "Chassaing_Bioreactors"
author: "Kelsey Huus"
date: "2025-10-14"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
##Packages
library(dplyr)
library(ggplot2)
#library(data.table)
library(tidyr)
library(Maaslin2)

```

```{r metadata}
#mouse metadata #to see if I can use the same samples for bioreactor experiments
mouse_meta <- read.csv("/ebio/abt3_projects2/uHEAT/data/Mouse/MouseMicrobiome/wt_fmt_metadata.csv")
mouse_meta
mouse_meta$Participant_ID <- sapply(strsplit(mouse_meta$SampleID,"_"), `[`, 1)
mouse_meta$Participant_ID <- gsub("H", "HEAT_", mouse_meta$Participant_ID)

#per-sample metadata #note: mocks, blanks & samples with low read depth already excluded #so are old & infected samples
#updated: this version also excludes SERIES 2 of all double series.
meta_seq <- read.csv("/ebio/abt3_projects2/uHEAT/data/metadata/meta_seq_clean_series1.24.08.18.csv", 
                     row.names=1)
meta_seq$Bristol_Stool_Score2 <- ifelse(meta_seq$Bristol_Stool_Score2=="bNA", yes=NA, no=meta_seq$Bristol_Stool_Score2) #correct BSS label error


#number of clean samples
length(meta_seq$SampleID)

#per-person metadata, matching
meta <- meta_seq %>% filter(!duplicated(Participant_ID)&!is.na(fever7d_aboveavg)) %>%
  dplyr::select(-c(SampleID, Relative_Day, Time0, Time01))
dim(meta)

#combine metadata
mouse_meta <- mouse_meta %>% select(-c("SampleID")) %>% full_join(meta) %>%
  filter(!is.na(Mice.ID_Tube))
mouse_meta$Mice.ID_Tube <- paste("X", mouse_meta$Mice.ID_Tube, sep="") #make it compatible with bracken

#import the cleaned data tables generated by "DNA.Maaslin.Executable.R"
feces_df_genus <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_input/DNA_feces_df_genus.csv", row.names=1)
feces_df_genus <- feces_df_genus[,order(names(feces_df_genus))] #order
meta_seq <- meta_seq[order(meta_seq$SampleID),] #order
identical(names(feces_df_genus), meta_seq$SampleID)

```

```{r taxa}
#check taxa present in the original human donor sample i.e. Waltera / Acetatifactor / Lachno
plot_gen <- data.frame(t(feces_df_genus), meta_seq)
waltera <- plot_gen %>% group_by(Participant_ID) %>%
  summarize(Waltera_mean=mean(Acetatifactor),
            Waltera_median=median(Acetatifactor),
            Waltera_sd=sd(Acetatifactor),
            Waltera_min=min(Acetatifactor),
            Waltera_max=max(Acetatifactor))
#general Waltera distribution
summary(waltera$Waltera_mean) #0.0002 - 0.05, mean 0.0098

bottom_waltera <- head(waltera[order(waltera$Waltera_mean),])
top_waltera <- head(waltera[order(waltera$Waltera_mean, decreasing = TRUE),])

donor_waltera <- waltera %>% filter(Participant_ID %in% mouse_meta$Participant_ID)
#any overlap with the top and bottom? probably not...
intersect(donor_waltera$Participant_ID, top_waltera$Participant_ID) #none
intersect(donor_waltera$Participant_ID, bottom_waltera$Participant_ID) #H074 is represented and has very low Waltera

donor_waltera[order(donor_waltera$Waltera_mean, decreasing=TRUE),] #H134 has quite a high abundance at 0.0224 i.e. 2.24%
#I think it is worth it to aliquot H157 which has even higher, and consistently? do I have H157? I do, also H096

potential_donors <- c("HEAT_157", "HEAT_096", #high
                      "HEAT_074", "HEAT_081") #low
bioreactor_donors <- meta %>% filter(Participant_ID %in% potential_donors) %>% select(Participant_ID,
                                                                 fever7d, fever7d_aboveavg, Sex, Age,
                                                                 previous_COVID19_infection, Vaccine1_Type,
                                                                 Vaccine2_Type, Vaccine3_Type, StudyVaccine_Type,
                                                                 Vaccine_StudyDose, COVID_infection_during_study) %>%
  full_join(waltera) %>%
  filter(!is.na(Age))

#write.csv(bioreactor_donors, "/ebio/abt3_projects2/uHEAT/data/ChassaingCMC/Bioreactors/bioreactor_potentialdonors.csv")

#check also at species level: is there A. muris present in ÂµHEAT?
feces_df_species <- read.csv("/ebio/abt3_projects2/uHEAT/code/R_code/uHEAT_CleanMarkdowns_Diet3/maaslin2_input/DNA_feces_df_species.csv", row.names=1)
waltera_sp <- feces_df_species[grep("Acetatifactor", row.names(feces_df_species)),]
waltera_sp <- as.data.frame(t(waltera_sp))
summary(waltera_sp$`Acetatifactor muris`) #at very very low abundance

```

```{r 16S data import}
#16S data from Chassaing Bioreactor experiment
otu <- read.csv("/ebio/abt3_projects2/uHEAT/data/ChassaingCMC/Bioreactors/16S_otu_table.csv",
                row.names=1)
#extract a small meta file
meta16S <- data.frame(Sample=otu$Donor_Timepoint_treatment,
                      Donor=sapply(strsplit(otu$Donor_Timepoint_treatment,"_"), `[`, 1),
                      Diet=sapply(strsplit(otu$Donor_Timepoint_treatment,"_"), `[`, 2),
                      Timepoint=sapply(strsplit(otu$Donor_Timepoint_treatment,"_"), `[`, 3))
meta16S$Time_days <- as.numeric(meta16S$Timepoint) -3
meta16S <- meta16S %>%
  dplyr::group_by(Sample) %>%
  dplyr::mutate(Rep = dplyr::row_number()) %>%
  dplyr::ungroup()
meta16S$SampleID <- paste(meta16S$Sample, meta16S$Rep, sep="_")
meta16S$Time_intervention <- ifelse(meta16S$Time_days<0, "Before", "After")

#convert to rel abund
identical(meta16S$Sample, otu$Donor_Timepoint_treatment)
row.names(otu) <- meta16S$SampleID
otu$Donor_Timepoint_treatment <- NULL
otu_ra <- otu/rowSums(otu)

#filter out very low abundance taxa
dim(otu_ra) #288 samples, 992 taxa
otu_ra_filt <- otu_ra[,which(colSums(otu_ra==0)<(length(row.names(otu_ra))*0.9))] #less than 90% zeroes
dim(otu_ra_filt) #195 prevalent(ish) taxa

#renormalize the filtered table
otu_ra_filt <- otu_ra_filt/rowSums(otu_ra_filt)

#make a tax table
tax <- data.frame(Taxon=names(otu_ra_filt),
                  Phylum=sapply(strsplit(names(otu_ra_filt),"p_"), `[`, 2),
                  Order=sapply(strsplit(names(otu_ra_filt),"o_"), `[`, 2),
                  Family=sapply(strsplit(names(otu_ra_filt),"f_"), `[`, 2),
                  Genus=sapply(strsplit(names(otu_ra_filt),"g_"), `[`, 2),
                  Species=sapply(strsplit(names(otu_ra_filt),"s_"), `[`, 2))
tax$Order <- sapply(strsplit(tax$Order,".", fixed=TRUE), `[`, 1)
tax$Family <- sapply(strsplit(tax$Family,".", fixed=TRUE), `[`, 1)
tax$Genus <- sapply(strsplit(tax$Genus,".", fixed=TRUE), `[`, 1)
tax$Species <- sapply(strsplit(tax$Species,".", fixed=TRUE), `[`, 1)
tax$Phylum <- sapply(strsplit(tax$Phylum,".", fixed=TRUE), `[`, 1)

#genus-level otu table.
identical(names(otu_ra_filt), tax$Taxon)
otu_g <- as.data.frame(t(otu_ra_filt))
otu_g$Genus <- gsub("_", "", tax$Genus, fixed=TRUE)
otu_g$Genus <- ifelse(is.na(otu_g$Genus), yes=paste(tax$Order, tax$Family, "g", sep="."), no=otu_g$Genus)
otu_g$Genus <- ifelse(otu_g$Genus==""|otu_g$Genus=="NA.NA.g", yes=paste(tax$Phylum, "g", sep="."), no=otu_g$Genus)
otu_g$Genus <- gsub("_", "", otu_g$Genus, fixed=TRUE)
otu_g <- otu_g %>% group_by(Genus) %>%
  summarize_all(sum)
dim(otu_g) #140 genera
colSums(otu_g[,c(2:length(names(otu_g)))]) #some rounding error (?) but basically 1

#family-level otu table
identical(names(otu_ra_filt), tax$Taxon)
otu_f <- as.data.frame(t(otu_ra_filt))
otu_f$Family <- gsub("_", "", tax$Family, fixed=TRUE)
otu_f$Family <- ifelse(is.na(otu_f$Family), yes=paste(gsub("_", "", tax$Phylum), "f", sep="."), no=otu_f$Family)
otu_f <- otu_f %>% group_by(Family) %>%
  summarize_all(sum)
dim(otu_f) #46 families

```

```{r 16S data analyze}
#are there any genera (besides Waltera/Acetatifactor) that increase post-vaccine in the CMC or additives group?
#use Maaslin2
otu_g$Genus[1] <- c("Unknown")
otu_g$Genus[which(is.na(otu_g$Genus))] <- c("Unknown2")
otu_g <- as.data.frame(otu_g)
row.names(otu_g) <- otu_g$Genus
otu_g$Genus <- NULL
meta16S <- as.data.frame(meta16S)
row.names(meta16S) <- meta16S$SampleID

identical(names(otu_g), row.names(meta16S))

##Genus level
fit_data = Maaslin2(
  input_data = otu_g, 
  input_metadata = meta16S, 
  output = "maaslin2_output", 
  fixed_effects = c("Time_intervention", "Diet"),
  random_effects = c("Sample"), #like participant ID - replicates over time
  reference=c("Diet", "Ctrl"), #tell it to use ctrl as reference
  normalization="tss", #total sum scaling
  min_abundance=0.0001, #min abundance 10^-3; anything lower gets unbelievable
  min_prevalence=0.25, #must be present in 25% of samples
  max_significance=0.05, #calm down with the non significant associations and plots
  plot_scatter=FALSE
)
diet_g <- fit_data$results
diet_g %>% dplyr::filter(qval<0.05&metadata=="Time_intervention") 
diet_g %>% dplyr::filter(qval<0.05&metadata=="Diet") 

#woo, Acetatifactor is one of the top hits!! also 2x Clostridia in the CMC condition, and Anaerobutyricum in Cocktail

#temporally it is a bit confusing, I think it would be better to analyze only the 'after' samples.
meta16S_after <- meta16S %>% filter(Time_intervention=="After")
otu_g_after <- otu_g[,which(names(otu_g) %in% meta16S_after$SampleID)]
identical(names(otu_g_after), row.names(meta16S_after))

##Genus level
fit_data = Maaslin2(
  input_data = otu_g_after, 
  input_metadata = meta16S_after, 
  output = "maaslin2_output", 
  fixed_effects = c("Diet"),
  random_effects = c("Sample"), #like participant ID - replicates over time
  reference=c("Diet", "Ctrl"), #tell it to use ctrl as reference
  normalization="tss", #total sum scaling
  min_abundance=0.0001, #min abundance 10^-3; anything lower gets unbelievable
  min_prevalence=0.25, #must be present in 25% of samples
  max_significance=0.05, #calm down with the non significant associations and plots
  plot_scatter=FALSE
)
diet_g_after <- fit_data$results
diet_g_after %>% dplyr::filter(qval<0.05&metadata=="Diet") 

tax$Genus <- gsub("_", "", tax$Genus, fixed=TRUE)

diet_g_after <- diet_g_after %>% filter(qval<0.05&metadata=="Diet") %>%
  rename(Genus=feature) %>%
  full_join(tax)

#write.csv(diet_g_after, "/ebio/abt3_projects2/uHEAT/data/ChassaingCMC/Bioreactors/maaslin_results_diet_g_after.csv")
  
##Family level?
otu_f$Genus[which(is.na(otu_f$Genus))] <- c("Unknown2")
otu_f <- as.data.frame(otu_f)
row.names(otu_f) <- otu_f$Genus
otu_f$Genus <- NULL

otu_f_after <- otu_f[,which(names(otu_f) %in% meta16S_after$SampleID)]
identical(names(otu_f_after), row.names(meta16S_after))

fit_data = Maaslin2(
  input_data = otu_f_after, 
  input_metadata = meta16S_after, 
  output = "maaslin2_output", 
  fixed_effects = c("Diet"),
  random_effects = c("Sample"), #like participant ID - replicates over time
  reference=c("Diet", "Ctrl"), #tell it to use ctrl as reference
  normalization="tss", #total sum scaling
  min_abundance=0.0001, #min abundance 10^-3; anything lower gets unbelievable
  min_prevalence=0.25, #must be present in 25% of samples
  max_significance=0.05, #calm down with the non significant associations and plots
  plot_scatter=FALSE
)
diet_f_after <- fit_data$results
diet_f_after %>% dplyr::filter(qval<0.05&metadata=="Diet") #technically yes, Lachno and Rumino both increase on the cocktail 

#write out results.
#write.csv(diet_f_after, "/ebio/abt3_projects2/uHEAT/data/ChassaingCMC/Bioreactors/maaslin_results_diet_f_after.csv")

```

```{r 16S data results}
diet_g_after <- read.csv("/ebio/abt3_projects2/uHEAT/data/ChassaingCMC/Bioreactors/maaslin_results_diet_g_after.csv")
diet_g_after
#Alongside Acetatifactor (which is one of the top hits)
#we have blooms in Anaerobutyricum (Lachno, non-motile), Hominenteromicrobium (Oscillo, I think non-motile), Gemmiger (! Oscillospirales, non-motile), Agathobacter (probably motile), Eisenbergiella
#check taxonomy because the internet is confused

```

```{r barplot genus}
#and, at genus level?
otu_g$Genus
#clean up a few of the taxa which have unnecessary orders attached and will come up later (bad fix, sorry)
otu_g$Genus <- gsub("Clostridiales.Clostridiaceae222000.g", "Clostridiaceae222000.g", otu_g$Genus, fixed=TRUE)
otu_g$Genus <- gsub("Peptostreptococcales.Peptostreptococcaceae256921.g", "Peptostreptococcaceae256921.g", otu_g$Genus, fixed=TRUE)
otu_g$Genus <- gsub("Lactobacillales.Enterococcaceae.g", "Enterococcaceae.g", otu_g$Genus, fixed=TRUE)


taxa_to_keep <- data.frame(Genus=otu_g$Genus, abund=rowSums(otu_g[,c(2:length(names(otu_g)))]))
taxa_to_keep[order(taxa_to_keep$abund, decreasing=TRUE),] #Acetatifactor is relatively minor overall

top_taxa <- (head(taxa_to_keep[order(taxa_to_keep$abund, decreasing=TRUE),], 25))$Genus #the other 2 are so minor
top_taxa

barplot_genus <- otu_g %>% pivot_longer(cols=c(2:length(names(otu_g))), 
                                        names_to="SampleID", values_to="Abundance")

barplot_genus$plot_taxa <- ifelse(barplot_genus$Genus %in% top_taxa,
                                     yes=barplot_genus$Genus,
                                     no="others")

bar_plot <- barplot_genus %>% group_by(SampleID, plot_taxa) %>% summarize(Abundance=sum(Abundance))


#put the taxa in order so that "others" is last
top_taxa_order <- c(top_taxa, "others")
bar_plot$plot_taxa <- factor(bar_plot$plot_taxa,
                               levels=top_taxa_order)

#add metadata
bar_plot <- bar_plot %>%
  full_join(meta16S)

#basic bar plot
bar_plot$Diet <- factor(bar_plot$Diet, levels=c("Ctrl", "CMC", "Cocktail"))

bar_plot_before <- bar_plot %>% filter(Time_intervention=="Before")
bar_plot_after <- bar_plot %>% filter(Time_intervention=="After")

#group 3 replicate bioreactors per sample #both baseline and after
bar_plot_simple <- bar_plot %>% group_by(Sample, plot_taxa, Donor, Diet, Timepoint, Time_days, Time_intervention) %>%
  summarize(Abundance=mean(Abundance))
#make a slightly simpler sample id, since it will be grouped by diet anyways.
bar_plot_simple$Sample2 <- gsub("Ctrl_", "", bar_plot_simple$Sample)
bar_plot_simple$Sample2 <- gsub("CMC_", "", bar_plot_simple$Sample2)
bar_plot_simple$Sample2 <- gsub("Cocktail_", "", bar_plot_simple$Sample2)

#versus: make an ungrouped version, so it can be shown by time, too. #this one by sample ID
#also exclude time from this sample id, so it stacks...
bar_plot$SampleID2 <- paste(bar_plot$Donor, bar_plot$Rep, sep="_")
#this version is nice (Time_days ~ Diet) but I think it's too big/complicated.

#versus: ungrouped, but facet by inoculum / donor. #this is maybe the most informative
bar_plot$SampleID3 <- paste(bar_plot$Timepoint, bar_plot$Rep, sep="_")

#this plot is nice but it doesn't fit. Go back to the original.
#pdf(width=15, height=15, "/ebio/abt3_projects2/uHEAT/data/RevisionPlots/S#_BioreactorBarplot_genus_v3.pdf")
p <- ggplot(bar_plot, aes(SampleID3, Abundance, fill=plot_taxa)) + 
  geom_bar(stat="identity", colour="black") +
  facet_grid(Donor ~ Diet, drop=FALSE,scale="free",space="free_x") + 
  theme_bw() +
  labs(x="Sample", y="Proportion") + 
  scale_y_continuous(expand = c(0,0)) + theme(strip.background = element_rect(fill="gray85")) +
  scale_fill_manual(values = colours) + theme(panel.spacing = unit(0.3, "lines")) + 
  theme(axis.text.x=element_text(angle=90,hjust=0.5,vjust=0.5)) + 
  guides(fill=guide_legend(title="Taxa")) +
  xlab(NULL)
p
#dev.off()

#pdf(width=15, height=5, "/ebio/abt3_projects2/uHEAT/data/RevisionPlots/S#_BioreactorBarplot_genus_v1.pdf")
p <- ggplot(bar_plot_simple, aes(Sample2, Abundance, fill=plot_taxa)) + 
  geom_bar(stat="identity", colour="black") +
  facet_grid(. ~ Diet, drop=FALSE,scale="free",space="free_x") + 
  theme_bw() +
  labs(x="Sample", y="Proportion") + 
  scale_y_continuous(expand = c(0,0)) + theme(strip.background = element_rect(fill="gray85")) +
  scale_fill_manual(values = colours) + theme(panel.spacing = unit(0.3, "lines")) + 
  theme(axis.text.x=element_text(angle=90,hjust=0.5,vjust=0.5)) + 
  guides(fill=guide_legend(title="Taxa")) +
  xlab("Sample (Inoculum_Timepoint)")
p
#dev.off()

#quite a strong bloom of Agathobacter in the bioreactors that corresponds to fecal flagellin levels

```

```{r barplot family}

#and, at family level
otu_f$Family
taxa_to_keep <- data.frame(Family=otu_f$Family, abund=rowSums(otu_f[,c(2:length(names(otu_f)))]))
taxa_to_keep[order(taxa_to_keep$abund, decreasing=TRUE),] #Acetatifactor is relatively minor overall
top_taxa <- (head(taxa_to_keep[order(taxa_to_keep$abund, decreasing=TRUE),], 25))$Family
top_taxa

barplot_fam <- otu_f %>% pivot_longer(cols=c(2:length(names(otu_f))), 
                                        names_to="SampleID", values_to="Abundance")

barplot_fam$plot_taxa <- ifelse(barplot_fam$Family %in% top_taxa,
                                     yes=barplot_fam$Family,
                                     no="others")

bar_plot <- barplot_fam %>% group_by(SampleID, plot_taxa) %>% summarize(Abundance=sum(Abundance))


#put the taxa in order so that "others" is last
top_taxa_order <- c(top_taxa, "others")
bar_plot$plot_taxa <- factor(bar_plot$plot_taxa,
                               levels=top_taxa_order)

#add metadata
bar_plot <- bar_plot %>%
  full_join(meta16S)

#basic bar plot
bar_plot$Diet <- factor(bar_plot$Diet, levels=c("Ctrl", "CMC", "Cocktail"))

bar_plot_before <- bar_plot %>% filter(Time_intervention=="Before")
bar_plot_after <- bar_plot %>% filter(Time_intervention=="After")

#group 3 replicate bioreactors per sample
bar_plot_simple <- bar_plot_after %>% group_by(Sample, plot_taxa, Donor, Diet, Timepoint, Time_days, Time_intervention) %>%
  summarize(Abundance=mean(Abundance))

#pdf(width=15, height=5, "/ebio/abt3_projects2/uHEAT/data/RevisionPlots/S#_Bioreactorbarplot_fam.pdf")
p <- ggplot(bar_plot_simple, aes(Sample, Abundance, fill=plot_taxa)) + 
  geom_bar(stat="identity", colour="black") +
  facet_grid(. ~ Diet, drop=FALSE,scale="free",space="free_x") + 
  theme_bw() +
  labs(x="Sample", y="Proportion") + 
  scale_y_continuous(expand = c(0,0)) + theme(strip.background = element_rect(fill="gray85")) +
  scale_fill_manual(values = colours) + theme(panel.spacing = unit(0.3, "lines")) + 
  theme(axis.text.x=element_text(angle=90,hjust=0.5,vjust=0.5)) + 
  guides(fill=guide_legend(title="Taxa"))
p
#dev.off()

#quite a strong bloom of Agathobacter in the bioreactors that corresponds to fecal flagellin levels

```

